<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2025-01-01">
<meta name="description" content="Compendium of recent articles, resources, and blogs in the area of medicine discovery">

<title>Medicine drug discovery resources – Omnis tempus datum</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark-d166b450ba5a8e9f7a0ab969bf6592c1.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-6bd9cfa162949bde0a231f530c97869d.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark-638ce43a2b8cb8132871590242e342ae.min.css" rel="prefetch" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Omnis tempus datum</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About me</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/pgg1610"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/mepgg"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Medicine drug discovery resources</h1>
                  <div>
        <div class="description">
          Compendium of recent articles, resources, and blogs in the area of medicine discovery
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">chemical-science</div>
                <div class="quarto-category">machine-learning</div>
                <div class="quarto-category">resources</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">January 1, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#noteworthy-blogs-to-follow" id="toc-noteworthy-blogs-to-follow" class="nav-link active" data-scroll-target="#noteworthy-blogs-to-follow">Noteworthy blogs to follow:</a></li>
  <li><a href="#online-resources" id="toc-online-resources" class="nav-link" data-scroll-target="#online-resources">Online resources</a></li>
  <li><a href="#books" id="toc-books" class="nav-link" data-scroll-target="#books">Books</a></li>
  <li><a href="#best-practices" id="toc-best-practices" class="nav-link" data-scroll-target="#best-practices">Best practices</a></li>
  <li><a href="#pharma-rd-business" id="toc-pharma-rd-business" class="nav-link" data-scroll-target="#pharma-rd-business">Pharma R&amp;D Business</a></li>
  <li><a href="#overviews-and-reviews" id="toc-overviews-and-reviews" class="nav-link" data-scroll-target="#overviews-and-reviews">Overviews and Reviews</a></li>
  <li><a href="#commentary" id="toc-commentary" class="nav-link" data-scroll-target="#commentary">Commentary</a></li>
  <li><a href="#industry-focused-drug-discovery-reviews" id="toc-industry-focused-drug-discovery-reviews" class="nav-link" data-scroll-target="#industry-focused-drug-discovery-reviews">Industry-focused drug discovery reviews</a></li>
  <li><a href="#special-journal-issues" id="toc-special-journal-issues" class="nav-link" data-scroll-target="#special-journal-issues">Special Journal Issues</a></li>
  <li><a href="#meeting-notes" id="toc-meeting-notes" class="nav-link" data-scroll-target="#meeting-notes">Meeting notes</a></li>
  <li><a href="#chemical-modalities" id="toc-chemical-modalities" class="nav-link" data-scroll-target="#chemical-modalities">Chemical modalities</a></li>
  <li><a href="#meta-themes-on-optimizing-small-molecules" id="toc-meta-themes-on-optimizing-small-molecules" class="nav-link" data-scroll-target="#meta-themes-on-optimizing-small-molecules">Meta themes on optimizing small molecules</a></li>
  <li><a href="#synthesis-chemistry" id="toc-synthesis-chemistry" class="nav-link" data-scroll-target="#synthesis-chemistry">Synthesis Chemistry</a></li>
  <li><a href="#structure-elucidation-analytical-techniques-modeling" id="toc-structure-elucidation-analytical-techniques-modeling" class="nav-link" data-scroll-target="#structure-elucidation-analytical-techniques-modeling">Structure elucidation / analytical techniques modeling</a></li>
  <li><a href="#large-chemical-libraries" id="toc-large-chemical-libraries" class="nav-link" data-scroll-target="#large-chemical-libraries">Large chemical libraries</a></li>
  <li><a href="#virtual-screeening" id="toc-virtual-screeening" class="nav-link" data-scroll-target="#virtual-screeening">Virtual screeening</a></li>
  <li><a href="#fragment-based-drug-discovery" id="toc-fragment-based-drug-discovery" class="nav-link" data-scroll-target="#fragment-based-drug-discovery">Fragment-based drug discovery</a></li>
  <li><a href="#protein-engineering" id="toc-protein-engineering" class="nav-link" data-scroll-target="#protein-engineering">Protein engineering</a></li>
  <li><a href="#protein-ligand-interactions" id="toc-protein-ligand-interactions" class="nav-link" data-scroll-target="#protein-ligand-interactions">Protein-ligand interactions</a>
  <ul class="collapse">
  <li><a href="#conformer-generators" id="toc-conformer-generators" class="nav-link" data-scroll-target="#conformer-generators">Conformer generators</a></li>
  <li><a href="#binding-energetic-prediction" id="toc-binding-energetic-prediction" class="nav-link" data-scroll-target="#binding-energetic-prediction">Binding energetic prediction</a></li>
  </ul></li>
  <li><a href="#molecular-dynamics" id="toc-molecular-dynamics" class="nav-link" data-scroll-target="#molecular-dynamics">Molecular-dynamics</a></li>
  <li><a href="#cheminformatics-focus" id="toc-cheminformatics-focus" class="nav-link" data-scroll-target="#cheminformatics-focus">Cheminformatics-focus</a>
  <ul class="collapse">
  <li><a href="#representation" id="toc-representation" class="nav-link" data-scroll-target="#representation">Representation</a></li>
  <li><a href="#predictive-modeling" id="toc-predictive-modeling" class="nav-link" data-scroll-target="#predictive-modeling">Predictive modeling</a></li>
  </ul></li>
  <li><a href="#pharmacophore-shape-searching" id="toc-pharmacophore-shape-searching" class="nav-link" data-scroll-target="#pharmacophore-shape-searching">Pharmacophore / shape searching</a>
  <ul class="collapse">
  <li><a href="#qsar-benchmarks" id="toc-qsar-benchmarks" class="nav-link" data-scroll-target="#qsar-benchmarks">QSAR benchmarks</a></li>
  <li><a href="#matched-molecular-pair" id="toc-matched-molecular-pair" class="nav-link" data-scroll-target="#matched-molecular-pair">Matched molecular-pair</a></li>
  <li><a href="#r-group-replacement-dataset" id="toc-r-group-replacement-dataset" class="nav-link" data-scroll-target="#r-group-replacement-dataset">R-group replacement dataset</a></li>
  <li><a href="#enumeration-of-chemical-space" id="toc-enumeration-of-chemical-space" class="nav-link" data-scroll-target="#enumeration-of-chemical-space">Enumeration of chemical space</a></li>
  </ul></li>
  <li><a href="#explainableinterpretable-machine-learning" id="toc-explainableinterpretable-machine-learning" class="nav-link" data-scroll-target="#explainableinterpretable-machine-learning">Explainable/Interpretable Machine Learning</a></li>
  <li><a href="#uncertainty-quantification" id="toc-uncertainty-quantification" class="nav-link" data-scroll-target="#uncertainty-quantification">Uncertainty quantification</a></li>
  <li><a href="#active-learning" id="toc-active-learning" class="nav-link" data-scroll-target="#active-learning">Active Learning</a>
  <ul class="collapse">
  <li><a href="#multi-parameter-optimization" id="toc-multi-parameter-optimization" class="nav-link" data-scroll-target="#multi-parameter-optimization">Multi-parameter optimization</a></li>
  </ul></li>
  <li><a href="#transfer-learning" id="toc-transfer-learning" class="nav-link" data-scroll-target="#transfer-learning">Transfer Learning</a></li>
  <li><a href="#meta-learning" id="toc-meta-learning" class="nav-link" data-scroll-target="#meta-learning">Meta Learning</a></li>
  <li><a href="#federated-learning" id="toc-federated-learning" class="nav-link" data-scroll-target="#federated-learning">Federated Learning</a></li>
  <li><a href="#generative-design" id="toc-generative-design" class="nav-link" data-scroll-target="#generative-design">Generative design</a></li>
  <li><a href="#computer-aided-synthesis-planning-casp" id="toc-computer-aided-synthesis-planning-casp" class="nav-link" data-scroll-target="#computer-aided-synthesis-planning-casp">Computer Aided Synthesis Planning (CASP)</a>
  <ul class="collapse">
  <li><a href="#data-driven-chemistry-modeling-and-reaction-optimization" id="toc-data-driven-chemistry-modeling-and-reaction-optimization" class="nav-link" data-scroll-target="#data-driven-chemistry-modeling-and-reaction-optimization">Data-driven chemistry modeling and reaction optimization</a></li>
  <li><a href="#automated-chemistry-workflows" id="toc-automated-chemistry-workflows" class="nav-link" data-scroll-target="#automated-chemistry-workflows">Automated chemistry workflows</a></li>
  <li><a href="#dna-encoded-libraries" id="toc-dna-encoded-libraries" class="nav-link" data-scroll-target="#dna-encoded-libraries">DNA-encoded Libraries</a></li>
  </ul></li>
  <li><a href="#large-language-models-llms" id="toc-large-language-models-llms" class="nav-link" data-scroll-target="#large-language-models-llms">Large Language Models (LLMs)</a></li>
  <li><a href="#data-extraction" id="toc-data-extraction" class="nav-link" data-scroll-target="#data-extraction">Data extraction</a></li>
  <li><a href="#code-packages" id="toc-code-packages" class="nav-link" data-scroll-target="#code-packages">Code / Packages:</a></li>
  <li><a href="#datasets-chemical-libraries" id="toc-datasets-chemical-libraries" class="nav-link" data-scroll-target="#datasets-chemical-libraries">Datasets &amp; Chemical libraries</a></li>
  <li><a href="#helpful-utilities" id="toc-helpful-utilities" class="nav-link" data-scroll-target="#helpful-utilities">Helpful utilities:</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<p>Last update: March 2025</p>
<section id="noteworthy-blogs-to-follow" class="level2">
<h2 class="anchored" data-anchor-id="noteworthy-blogs-to-follow">Noteworthy blogs to follow:</h2>
<p><strong>Cheminformatics</strong></p>
<ol type="1">
<li><p><a href="https://https://practicalcheminformatics.blogspot.com/">Patrick Walters Blog on Cheminformatics</a></p>
<ul>
<li><a href="https://github.com/PatWalters/practical_cheminformatics_tutorials">Cheminformatics Tutorials</a></li>
</ul></li>
<li><p><a href="https://iwatobipen.wordpress.com/">Is Life Worth Living</a></p></li>
<li><p><a href="https://whitead.github.io/dmol-book/intro.html">Andrew White’s Deep Learning for Molecule and Material</a></p></li>
<li><p><a href="http://www.cheminformania.com">Cheminformia</a></p></li>
<li><p><a href="https://depth-first.com">Depth-First</a></p></li>
<li><p><a href="https://baoilleach.blogspot.com/">Noel O’Blog</a></p></li>
</ol>
<p><strong>Fragment-based drug dicovery</strong></p>
<ol type="1">
<li><a href="http://practicalfragments.blogspot.com/">Practical Fragments</a></li>
</ol>
<p><strong>Medicinal chemistry</strong></p>
<ol type="1">
<li><p><a href="https://mcconnellsmedchem.com/">Darryl B McConnell’s Medchem blogpost</a></p></li>
<li><p><a href="https://www.cambridgemedchemconsulting.com/resources/">Chris Swaim’s MedChem blogpost</a></p></li>
</ol>
<p><strong>Computational chemistry</strong> 1. <a href="https://www.linkedin.com/in/gilles-ouvry-8b7b2b5/">Gilles Ouvry</a></p>
<p><strong>General field</strong></p>
<ol type="1">
<li><p><a href="http://www.drugdiscovery.net/2019/07/25/new-post1/">DrugDiscovery.NET - Andreas Bender</a></p></li>
<li><p><a href="https://drughunter.com">DrugHunter</a></p></li>
<li><p><a href="https://www.science.org/blogs/pipeline">Derek Lowe’s In the Pipeline</a></p></li>
</ol>
</section>
<section id="online-resources" class="level2">
<h2 class="anchored" data-anchor-id="online-resources">Online resources</h2>
<ul>
<li><p><a href="https://github.com/volkamerlab/TeachOpenCADD">Andrea Volkmer, TeachOpenCADD: a teaching platform for computer-aided drug design (CADD)</a></p></li>
<li><p><a href="https://github.com/PatWalters/practical_cheminformatics_tutorials">Patrick Walter’s Cheminformatics Tutorials</a></p></li>
<li><p><a href="https://www.macinchem.org/blog/files/fe66130c1da3375e46d0512e483eb901-2791.php?utm_source=pocket_mylist">Pat Walters’ RSC CICAG Open Source Tools for Chemistry</a>.<a href="https://www.youtube.com/watch?v=2ZjerAGS_IQ">Video</a>. <a href="https://github.com/PatWalters/chem_tutorial">Github</a></p></li>
<li><p><a href="https://github.com/iwatobipen/py4chemoinformatics">Pen’s Python cookbook for Cheminformatics</a></p></li>
<li><p><a href="https://bit.ly/2SxItoc">Chem LibreText collection from ACS Division of Chemical Education</a></p></li>
<li><p><a href="https://greglandrum.github.io/rdkit-blog/">RDkit blogpost from Greg Landrum</a></p></li>
<li><p><a href="https://bertiewooster.github.io/">Jeremy Monat’s blogpost</a></p></li>
</ul>
</section>
<section id="books" class="level2">
<h2 class="anchored" data-anchor-id="books">Books</h2>
<ul>
<li><p><a href="https://link.springer.com/book/10.1007/978-1-60761-839-3">Bajorath, 2011. Chemoinformatics and Computational Chemical Biology. Methods in Molecular Biology.</a></p></li>
<li><p><a href="https://link.springer.com/book/10.1007/978-1-0716-1787-8">Heifetz, Alexander. (Ed.) (2022). “Artificial Intelligence in Drug Design.”</a></p></li>
</ul>
</section>
<section id="best-practices" class="level2">
<h2 class="anchored" data-anchor-id="best-practices">Best practices</h2>
<ul>
<li><a href="https://www.nature.com/articles/s41570-022-00391-9">Bender, Andreas, et al.&nbsp;“Evaluation guidelines for machine learning tools in the chemical sciences.” Nature Reviews Chemistry (2022): 1-15.</a>. <a href="https://www.nature.com/articles/s41570-022-00391-9.epdf?sharing_token=c7ajAY6RtejsBwo7JKb_EdRgN0jAjWel9jnR3ZoTv0PlqwefS9tuSzOUSTdLlQspcQfdrrg6BP1js7fwhK8sXckoDuc05MIHU8Tf2mCEeVGilKBg5tdIz-htajojgKeP-9SZLLqCDAphBNd8VUtODVWYw0CXTg8CY1uUfyiB8zk%3D">Temporary SharedIt Link</a></li>
</ul>
<p>Nice account outlining guidelines for evaluating different AI/ML methodologies in molecular science. They propose a checklist of tests and best practices to assess the practicality and importance of different methodologies thereby providing a framework on how to evaluate plethora of ML workflows being proposed in different areas of chemical science. The basis for not overlooking the older non-ML method when evaluating the ‘new’ learning-based method, emphasis on model interpretation to translate the corrleation to chemical causality and finally</p>
<ul>
<li><a href="https://www.nature.com/articles/s41557-021-00716-z">Artrith, Nongnuch, et al.&nbsp;“Best practices in machine learning for chemistry.” Nature chemistry 13.6 (2021): 505-508.</a></li>
</ul>
<p>Set of rules, considerations, and caveats to keep in mind when designing ML model for chemical science. The authors propose a checklist when evaluating ML models, while intuitive at first, when lot of the new ML papers are scanned through that lens, you can identify the shortcommings of the proposed model. This checklist is especially helpful for those entering just entering the field.</p>
</section>
<section id="pharma-rd-business" class="level2">
<h2 class="anchored" data-anchor-id="pharma-rd-business">Pharma R&amp;D Business</h2>
<ul>
<li><p><a href="https://www.sciencedirect.com/science/article/pii/S1359644623002428?via%3Dihub">Schuhmacher, Alexander, et al.&nbsp;“Analysis of pharma R&amp;D productivity–a new perspective needed.” Drug Discovery Today (2023): 103726.</a></p></li>
<li><p><a href="https://www.nature.com/articles/nrd3078">Paul, Steven M., et al.&nbsp;“How to improve R&amp;D productivity: the pharmaceutical industry’s grand challenge.” Nature reviews Drug discovery 9.3 (2010): 203-214.</a></p></li>
</ul>
</section>
<section id="overviews-and-reviews" class="level2">
<h2 class="anchored" data-anchor-id="overviews-and-reviews">Overviews and Reviews</h2>
<ul>
<li><a href="https://pubs.rsc.org/en/content/articlelanding/2020/CS/C9CS00786E#fn1">F. Strieth-Kalthoff, F. Sandfort, M. H. S. Segler, and F. Glorius, Machine learning the ropes: principles, applications and directions in synthetic chemistry, Chem. Soc. Rev</a></li>
</ul>
<p>Pedagogical account of various machine learning techniques, models, representation schemes from perspective of synthetic chemistry. Covers different applications of machine learning in synthesis planning, property prediction, molecular design, and reactivity prediction</p>
<ul>
<li><p><a href="https://www.annualreviews.org/doi/abs/10.1146/annurev-biodatasci-122120-124216">Rodríguez-Pérez, Raquel, Filip Miljković, and Jürgen Bajorath. “Machine Learning in Chemoinformatics and Medicinal Chemistry.” Annual review of biomedical data science 5 (2022)</a></p></li>
<li><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00519-x">Mariia Matveieva &amp; Pavel Polishchuk. Benchmarks for interpretation of QSAR models</a>. <a href="https://github.com/ci-lab-cz/ibenchmark">Github</a>. <a href="https://patwalters.github.io/practicalcheminformatics/jupyter/ml/interpretability/2021/06/03/interpretable.html">Patrick Walter’s blog on the paper</a>.</p></li>
</ul>
<p>Paper outlining good practices for interpretating QSAR (Quantative Structure-Property Prediction) models. Good set of heuristics and comparison in the paper in terms of model interpretability. Create 6 synthetic datasets with varying complexity for QSAR tasks. The authors compare interpretability of graph-based methods to conventional QSAR methods. In regards to performance graph-based models show low interpretation compared to conventional QSAR method.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/full/10.1021/acs.accounts.0c00699">W. Patrick Walters &amp; Regina Barzilay. Applications of Deep Learning in Molecule Generation and Molecular Property Prediction</a></li>
</ul>
<p>Recent review summarising the state of the molecular property prediction and structure generation research. In spite of exciting recent advances in the modeling efforts, there is a need to generate better (realistic) training data, assess model prediction confidence, and metrics to quantify molecular generation performance.</p>
<ul>
<li><p><a href="https://chemrxiv.org/articles/preprint/Navigating_through_the_Maze_of_Homogeneous_Catalyst_Design_with_Machine_Learning/12786722/1">Navigating through the Maze of Homogeneous Catalyst Design with Machine Learning</a></p></li>
<li><p><a href="https://doi.org/10.1016/j.trechm.2020.11.004">Coley, C. W. Defining and Exploring Chemical Spaces. Trends in Chemistry 2020</a></p></li>
<li><p><a href="https://www.sciencedirect.com/science/article/pii/S0169409X21001800?via%3Dihub">Machine learning directed drug formulation development</a></p></li>
</ul>
<p>Review from Aspuru-Guzik and Allen’s group discussing how ML can be leveraged for various tasks in drug formulation tasks.</p>
</section>
<section id="commentary" class="level2">
<h2 class="anchored" data-anchor-id="commentary">Commentary</h2>
<ul>
<li><p><a href="https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/cmdc.202400074">Weaver, Donald F. “Chemists Invent Drugs and Drugs Save Lives.” ChemMedChem (2024): e202400074.</a></p></li>
<li><p><a href="https://www.ft.com/content/3e57ad6c-493d-4874-a663-0cb200d3cdb5">Will AI turbocharge the hunt for new drugs?</a></p></li>
<li><p><a href="https://practicalcheminformatics.blogspot.com/2023/02/generative-molecular-design-we-need-to.html">Comment about generative design from Patrick Walters</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41587-020-0418-2">Walters, W. Patrick, and Mark Murcko. “Assessing the impact of generative AI on medicinal chemistry.” Nature biotechnology 38.2 (2020): 143-145.</a></p></li>
</ul>
<p>Correspondence on assessing the impact of AI on medicinal chemistry. It is a well written account on practical implication of generative design on pharmaceutical research.They outline two recent cases of ‘success’ of AI generative design in drug discovery and give more context and propose best practices for furthering the development of algorithms and drug discovery pipelines.</p>
<ul>
<li><a href="https://practicalcheminformatics.blogspot.com/2023/08/we-need-better-benchmarks-for-machine.html">We need better benchmarking for machine learning in drug discovery</a></li>
</ul>
<p>Very good post outlining the focus on the good practices and lack thereof for consistent datasets for comparing different ML algorithms.</p>
</section>
<section id="industry-focused-drug-discovery-reviews" class="level2">
<h2 class="anchored" data-anchor-id="industry-focused-drug-discovery-reviews">Industry-focused drug discovery reviews</h2>
<ul>
<li><a href="https://pubs.acs.org/doi/epdf/10.1021/acs.jmedchem.2c00334?ref=article_openPDF">Goldman, B., Kearnes, S., Kramer, T., Riley, P., &amp; Walters, W. P. (2022). Defining levels of automated chemical design. Journal of medicinal chemistry, 65(10), 7073-7087.</a></li>
</ul>
<p>Group at Relay Therapeutics propose a framework to categorize automated chemical design paradigm - splitting it into generator and decision axes. They give good example of model systems where the machine generates and human chemist select and more recently machine generated and machine chosen designs. In of these discussion, it is evident we havent achieved the full automated execution of a design cycle.</p>
<ul>
<li><a href="https://www.annualreviews.org/doi/10.1146/annurev-pharmtox-040323-040828?url_ver=Z39.88-2003&amp;rfr_id=ori%3Arid%3Acrossref.org&amp;rfr_dat=cr_pub++0pubmed">Hasselgren, Catrin, and Tudor I. Oprea. “Artificial Intelligence for Drug Discovery: Are We There Yet?.” Annual Review of Pharmacology and Toxicology 64 (2024).</a> <a href="https://arxiv.org/abs/2307.06521">ArXiv</a></li>
</ul>
<p>Latest review of the field and application of AI technologies to various functions of drug discovery. In addition to providing a quick review of the main technology the author list different case studies where AI has proposed clinical candidates across different therapeutic areas. Yet, the need for better data, novelty estimation, and validation in wet lab limit the full scale deployment and accuracy of AI pipelines in drug discovery. In closing they also hint at the limitation of ML model training a single property with single structure, QSAR, while in reality the molecule can exist in different conformers something multi-instance learning (MIL) can address. ‘It is reasonable to assume that user expertise, bias, and time constraints play a significant role in early drug discovery, often more so than AI.’</p>
<ul>
<li><p><a href="https://www.nature.com/articles/d41573-022-00025-1">Jayatunga, Madura KP, et al.&nbsp;“AI in small-molecule drug discovery: A coming wave.” Nat. Rev.&nbsp;Drug Discov 21 (2022): 175-176.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.1c01580">Abramov, Yuriy A., Guangxu Sun, and Qun Zeng. “Emerging Landscape of Computational Modeling in Pharmaceutical Development.” Journal of Chemical Information and Modeling (2022).</a></p></li>
</ul>
<p>Overview of methods and scope of computational methods used in the drug development process.</p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.1c02106">Dragovich, Peter S., et al.&nbsp;“Small-Molecule Lead-Finding Trends across the Roche and Genentech Research Organizations.” Journal of Medicinal Chemistry (2022).</a></p></li>
<li><p><a href="https://www.sciencedirect.com/science/article/pii/S1359644620305274">A. Bender and I. Cortés-Ciriano, “Artificial intelligence in drug discovery: what is realistic, what are illusions? Part 1: Ways to make an impact, and why we are not there yet,” Drug Discov. Today, vol.&nbsp;26, no. 2, pp.&nbsp;511–524, 2021</a></p></li>
</ul>
</section>
<section id="special-journal-issues" class="level2">
<h2 class="anchored" data-anchor-id="special-journal-issues">Special Journal Issues</h2>
<ol type="1">
<li><a href="https://pubs.acs.org/page/achre4/data-science-meets-chemistry">Data Science Meets Chemistry</a></li>
</ol>
<p>This issue includes contributions that demonstrate the profound impact data science techniques have had in chemistry including chemical and materials synthesis, catalyst and materials design, and overhauling the models used in traditional theoretical or computational chemistry.</p>
<ol start="2" type="1">
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jmedchem.0c01077">Journal of Medicinal Chemistry compendium of AI in Drug discovery issue</a></p></li>
<li><p><a href="https://pubs.acs.org/page/achre4/data-science-meets-chemistry">Account of Chemical Research Special Issue on advances in data-driven chemistry research</a></p></li>
<li><p><a href="https://pubs.acs.org/toc/jcisd8/62/9">Special Issue on Reaction Informatics and Chemical Space, Journal of Chemical Information and Modeling (2022)</a></p></li>
</ol>
</section>
<section id="meeting-notes" class="level2">
<h2 class="anchored" data-anchor-id="meeting-notes">Meeting notes</h2>
<ul>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/611cf1a6ac8b499b36458d19">Warr, W. (2021). National Institutes of Health (NIH) Workshop on Reaction Informatics</a></p></li>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/60c75883bdbb89984ea3ada5">Warr, W. (2021). Report on an NIH Workshop on Ultralarge Chemistry Databases.</a></p></li>
</ul>
</section>
<section id="chemical-modalities" class="level2">
<h2 class="anchored" data-anchor-id="chemical-modalities">Chemical modalities</h2>
<ul>
<li><a href="https://pubs.acs.org/doi/pdf/10.1021/acsmedchemlett.9b00582">Blanco, Maria-Jesus, and Kevin M. Gardinier. “New chemical modalities and strategic thinking in early drug discovery.” ACS medicinal chemistry letters 11.3 (2020): 228-231.</a></li>
</ul>
<p>Overview of different chemical modalities currently at work to address different disease targets. The article addresses the small molecule medicinal chemists and how they can expand their outlook of small molecules to include other molecular entities when considering the angle of attack for different target engagement strategies. The authors offer a nice set of tools and thought process when selecting possible drug modalities for different target classes and what questions should be asked when zeroing in a possible mode of action.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.3c00603">Targeted Protein Degradation: Advances, Challenges, and Prospects for Computational Methods</a></li>
</ul>
</section>
<section id="meta-themes-on-optimizing-small-molecules" class="level2">
<h2 class="anchored" data-anchor-id="meta-themes-on-optimizing-small-molecules">Meta themes on optimizing small molecules</h2>
<ul>
<li><p><a href="https://www.sciencedirect.com/science/article/abs/pii/S0169409X02000030?via%3Dihub">Walters, W. Patrick, and Mark A. Murcko. “Prediction of ‘drug-likeness’.” Advanced drug delivery reviews 54.3 (2002): 255-271.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/jm020017n">Veber, Daniel F., et al.&nbsp;“Molecular properties that influence the oral bioavailability of drug candidates.” Journal of medicinal chemistry 45.12 (2002): 2615-2623.</a></p></li>
</ul>
<p>Retrospective analysis on factors influencing the bioavailability of drug candidates. Authors find rotatable bonds and polar surface area or hydrogen bond count (sum of donor and accpetors) found to be important predictors of good oral bioavailability. Compounds having &lt;10 rotatable bonds and &lt;140 A (or &lt; 12 hydrogen bonds) have good chances of being orally bioavailable.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.7b00717">DeGoey, David A., et al.&nbsp;“Beyond the rule of 5: lessons learned from AbbVie’s drugs and compound collection: miniperspective.” Journal of Medicinal Chemistry 61.7 (2017): 2636-2651.</a></li>
</ul>
<p>AB-MPS calculated using cLogD, the number of aromatic rings (nAr), and the number of rotatable bonds (nRotB) according to the formula AB-MPS = Abs(cLogD −3) + nAr + nRotB. The lower the AB-MPS score, the more likely the compound is to be absorbed, and a value of ≤14 is reported to predict a higher probability of oral absorption.</p>
<ul>
<li><a href="https://www.sciencedirect.com/science/article/pii/S1367593118300176#bib0030">Poongavanam, Vasanthanathan, Bradley C. Doak, and Jan Kihlberg. “Opportunities and guidelines for discovery of orally absorbed drugs in beyond rule of 5 space.” Current Opinion in Chemical Biology 44 (2018): 23-29.</a></li>
</ul>
<p>Hueristics for oral bioavailability of molecules that are violating the rule of 5. MW may reach up to approximately 1000 Da provided that TPSA increases proportionally up to 250 Å2. In contrast, cLogP and HBDs must be carefully controlled at high MW. Our lack of ability to predict compound conformations and flexibility is currently a hurdle that is critical to overcome to enable further prospective design in oral bRo5 space.</p>
<ul>
<li><p><a href="https://doi.org/10.1021/jm4017625">Taylor, R. D.; MacCoss, M.; Lawson, A. D. G. Rings in Drugs. J. Med. Chem. 2014, 57 (14), 5845–5859.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.1c01215">Subbaiah, Murugaiah AM, and Nicholas A. Meanwell. “Bioisosteres of the phenyl ring: Recent strategic applications in lead optimization and drug design.” Journal of Medicinal Chemistry 64.19 (2021): 14046-14128.</a></p></li>
</ul>
<p>Looks at biosteric replacements for the phenyl rings in the lead optimization phase. Phenyl rings results in improve potency but have poor solubility and lipophilicitty. Find biosteres can be used to improve them.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.1c00761">Ertl, Peter. “Magic Rings: Navigation in the Ring Chemical Space Guided by the Bioactive Rings.” Journal of Chemical Information and Modeling (2021).</a></li>
</ul>
<p>Analyze the nature of rings which appear in bioactive compounds. Ring systems are systematically extracted from one billion molecules and are analyzed to discover a structure or correlation in the bioactivity and type of rings. No simple set of structural descriptors separating active and inactive rings could be identified, the separation is best described by a neural network model taking into account a complex combination of many substructure features.</p>
<ul>
<li><a href="https://www.nature.com/articles/s41570-022-00451-0">Hartung, Ingo V., Bayard R. Huck, and Alejandro Crespo. “Rules were made to be broken.” Nature Reviews Chemistry 7.1 (2023): 3-4.</a></li>
</ul>
<p>Longitudinal analysis of physico-chemical properties for approved drugs in the clinic. They show that most of the drugs flout most of the Lipinski’s rule of 5 except the HBD which is always consistently less 4. In addition, they show that in recent times, by categorizing the drugs in different time-bound classes, the mean MW and HBA has increased but mean HBD has constantly stayed less than 2.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/jacsau.2c00415">Young, Robert J., et al.&nbsp;“The time and place for nature in drug discovery.” Jacs Au 2.11 (2022): 2400-2416.</a></li>
</ul>
<p>Critique on the paper, <a href="https://fbdd-lit.blogspot.com/2024/05/a-time-and-place-for-nature-in-drug.html">interesting take</a></p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.4c01345?ref=pdf">Property-Based Drug Design Merits a Nobel Prize</a></li>
</ul>
<p>Probably contentious topic but a good short review of the property-based optimization thought process for medicine discovery.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.6b01807">Pennington, Lewis D., and Demetri T. Moustakas. “The necessary nitrogen atom: a versatile high-impact design element for multiparameter optimization.” Journal of Medicinal Chemistry 60.9 (2017): 3552-3579.</a></li>
</ul>
<p>Good perspective highlighting the impact of replacing CH group with N in aromatic and heteraromatic ring systems on molecular and physiochemical properties that translate to improved pharmacological properties.</p>
</section>
<section id="synthesis-chemistry" class="level2">
<h2 class="anchored" data-anchor-id="synthesis-chemistry">Synthesis Chemistry</h2>
<p>Catalog of recent research articles that look at synthesis chemistry from a point of view of computational workflows, how traditional synthetic chemistry methods can be combined with informatics to augment drug discovery and synthesis processes.</p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acscatal.2c05159#.Y6oIfVMoYEs">Ruck, Rebecca T., Neil A. Strotman, and Shane W. Krska. “The Catalysis Laboratory at Merck: 20 Years of Catalyzing Innovation.” ACS Catalysis 13 (2022): 475-503.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.accounts.0c00760">Dreher, Spencer D., and Shane W. Krska. “Chemistry informer libraries: Conception, early experience, and role in the future of cheminformatics.” Accounts of Chemical Research 54.7 (2021): 1586-1596.</a></p></li>
</ul>
<p>Curated set of substrates to quickly assess the practicality of synthetic methods with the complete capture of success and failure, that can optimize reaction conditions with a broader scope with respect to relevant applications.</p>
<ul>
<li><p><a href="https://www.science.org/doi/pdf/10.1126/science.aat0805">Campos, Kevin R., et al.&nbsp;“The importance of synthetic chemistry in the pharmaceutical industry.” Science 363.6424 (2019): eaat0805.</a></p></li>
<li><p><a href="https://www.science.org/doi/10.1126/science.add1383">Late-stage diversification of indole skeletons through nitrogen atom insertion</a></p></li>
</ul>
</section>
<section id="structure-elucidation-analytical-techniques-modeling" class="level2">
<h2 class="anchored" data-anchor-id="structure-elucidation-analytical-techniques-modeling">Structure elucidation / analytical techniques modeling</h2>
<ul>
<li><a href="https://arxiv.org/abs/2502.09571"></a></li>
</ul>
<p>DiffMS is a novel machine learning framework for generating molecular structures from mass spectra. It uses a formula-restricted encoder-decoder network, achieving state-of-the-art performance. The novelty lies in its discrete graph diffusion model and extensive pretraining. However, gaps remain in fully elucidating exact molecular structures due to inherent ambiguities in mass spectra.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/jacs.4c08300">Li, Haote, et al.&nbsp;“Rapid Quantification of Protein Secondary Structure Composition from a Single Unassigned 1D 13C Nuclear Magnetic Resonance Spectrum.” Journal of the American Chemical Society 146.40 (2024): 27542-27554.</a></li>
</ul>
<p>Secondary protein structure determination from 1D NMR without chemical shift assignments. Each residue is modeled as ensemble of secondary structure viz.&nbsp;alpha-helix, beta-sheet, random coil. While the approach can hint at residue-level differences, its primary output is the overall ensemble distribution of secondary structural elements.</p>
</section>
<section id="large-chemical-libraries" class="level2">
<h2 class="anchored" data-anchor-id="large-chemical-libraries">Large chemical libraries</h2>
<p>Over the past few years several entites offering ultra-large ensembles of chemical libraries which can be made on-demand or purchased immediately have emerged. The existence of such services has reinvigorated the field of virtual screening and combinatorial library design. In addition, research groups have devised novel ways to navigate these libraries, more efficiently and also understand the differences in the chemical space these library cover. Following are some of the key papers in the field.</p>
<ul>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/60c75883bdbb89984ea3ada5">Warr, W. (2021). Report on an NIH Workshop on Ultralarge Chemistry Databases.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.2c00224">Warr, Wendy A., et al.&nbsp;“Exploration of ultralarge compound collections for drug discovery.” Journal of Chemical Information and Modeling 62.9 (2022): 2021-2034.</a></p></li>
<li><p><a href="https://www.sciencedirect.com/science/article/pii/S1359644618304471">The next level in chemical space navigation: going far beyond enumerable compound libraries</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.1c01378?casa_token=Lo0YQQrtkcAAAAAA%3AhKiuAJ0GpAhXiNC-2cVO6Ixv0aCnf0J7mrr03mOxpCgH8OpVhaCnDxAkHPNvfm36VsDl8NxvtzusqQ">Bellmann, Louis, et al.&nbsp;“Comparison of combinatorial fragment spaces and its application to ultralarge make-on-demand compound catalogs.” Journal of Chemical Information and Modeling 62.3 (2022): 553-566.</a></p></li>
</ul>
<p>SpaceCompare: calculation of the overlap of large, nonenumerable combinatorial fragment spaces, utilizes topological fingerprints and the combinatorial character of these chemical spaces. Enamine’s REAL Space, WuXi’s GalaXi Space, and Otava’s CHEMriya. The overlap of the commercial make-on-demand catalogs is only in the low single-digit percent range, despite their large overall size.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.9b00367">Konze, Kyle D., et al.&nbsp;“Reaction-based enumeration, active learning, and free energy calculations to rapidly explore synthetically tractable chemical space and optimize potency of cyclin-dependent kinase 2 inhibitors.” Journal of chemical information and modeling 59.9 (2019): 3782-3793.</a></li>
</ul>
<p>PathFinder uses retrosynthetic analysis followed by combinatorial synthesis to generate novel compounds in synthetically accessible chemical space.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.0c00675">Irwin, John J., et al.&nbsp;“ZINC20—a free ultralarge-scale chemical database for ligand discovery.” Journal of chemical information and modeling 60.12 (2020): 6065-6073.</a></li>
</ul>
<p>New version of ZINC with two major new features: billions of new molecules and new methods to search them. As a fully enumerated database, ZINC can be searched precisely using explicit atomic-level graph-based methods. Over 97% of the core Bemis–Murcko scaffolds in make-on-demand libraries are unavailable from “in-stock” collections. Correspondingly, the number of new Bemis–Murcko scaffolds is rising almost as a linear fraction of the elaborated molecules. Thus, an 88-fold increase in the number of molecules in the make-on-demand versus the in-stock sets is built upon a 16-fold increase in the number of Bemis–Murcko scaffolds. The make-on-demand library is also more structurally diverse than physical libraries</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acsmedchemlett.3c00021">Neumann, Alexander, Lester Marrison, and Raphael Klein. “Relevance of the Trillion-Sized Chemical Space “eXplore” as a Source for Drug Discovery.” ACS Medicinal Chemistry Letters (2023).</a></li>
</ul>
<p>The authors examine the composition of the recently published and, so far, biggest chemical space, “eXplore”, which comprises approximately 2.8 trillion virtual product molecules. The utility of eXplore to retrieve interesting chemistry around approved drugs and common Bemis Murcko scaffolds has been assessed with several methods (FTrees, SpaceLight, SpaceMACS). Further, the overlap between several vendor chemical spaces and a physicochemical property distribution analysis has been performed. Despite the straightforward chemical reactions underlying its setup, eXplore is demonstrated to provide relevant and, most importantly, easily accessible molecules for drug discovery campaigns.</p>
<ul>
<li><a href="https://arxiv.org/pdf/2304.05386.pdf">Medina, Jorge, and Andrew D. White. “Bloom filters for molecules.” arXiv preprint arXiv:2304.05386 (2023).</a></li>
</ul>
<p>This paper proposes and studies Bloom filters for testing if a molecule is present in a set using either string or fingerprint representations. Bloom filters are small enough to hold billions of molecules in just a few GB of memory and check membership in sub-milliseconds. The authors found string representations can have a false positive rate below 1% and require significantly less storage than using fingerprints. Canonical SMILES with Bloom filters with the simple FNV hashing function provide fast and accurate membership tests with small memory requirements. They provide a general implementation and specific filters for detecting if a molecule is purchasable, patented, or a natural product according to existing databases at https://github.com/whitead/molbloom.</p>
</section>
<section id="virtual-screeening" class="level2">
<h2 class="anchored" data-anchor-id="virtual-screeening">Virtual screeening</h2>
<ul>
<li><p><a href="https://www.nature.com/articles/s43588-025-00777-x">Luttens, A., de Vaca, I. C., Sparring, L., Norinder, U., &amp; Carlsson, J. (2023). Rapid traversal of ultralarge chemical space using machine learning guided docking screens.</a></p></li>
<li><p><a href="https://pubs.rsc.org/en/content/articlelanding/2024/sc/d3sc04653b">Dodds, Michael, et al.&nbsp;“Sample efficient reinforcement learning with active learning for molecular design.” Chemical Science 15.11 (2024): 4146.</a></p></li>
</ul>
<p>An active learning system linked with an RL model (RL–AL) for molecular design, which aims to improve the sample-efficiency of the optimization process.</p>
<p>AZ group looking at generative design + RL + Virtual screening campaign</p>
<ul>
<li><a href="https://arxiv.org/pdf/2402.08210.pdf">Vakili, Mohammad Ghazi, et al.&nbsp;“Quantum Computing-Enhanced Algorithm Unveils Novel Inhibitors for KRAS.” arXiv preprint arXiv:2402.08210 (2024).</a></li>
</ul>
<p>First paper to showcase deployment of quantum-based generative models with virtual screening workflow on a target-based compound discovery. Chemistry42 is used for reward function implementation. The authors show that molecule generated from this combined effort are ‘better’ quality-wise than traditional workflow (LSTM and Genetic algorithm) and the modeling success downstream is roughly correlated with number of qubits employed. This is exciting more from technical standpoint of combining quantum + traditional workflows.</p>
<ul>
<li><a href="https://www.nature.com/articles/s41586-023-05905-z">Sadybekov, Anastasiia V., and Vsevolod Katritch. “Computational approaches streamlining drug discovery.” Nature 616.7958 (2023): 673-685.</a></li>
</ul>
<p>Nice review on virtual screening workflow for streamlining drug discovery</p>
<ul>
<li><a href="https://www.nature.com/articles/s41586-020-2117-z">Gorgulla, Christoph, et al.&nbsp;“An open-source drug discovery platform enables ultra-large virtual screens.” Nature 580.7805 (2020): 663-668.</a></li>
</ul>
<p>VirtualFlow as a tool for conducting virtual screening. The authors use VirtualFlow, to prepare one of the largest and freely available ready-to-dock ligand libraries, with more than 1.4 billion commercially available molecules. They screening ~1 billion compounds and identified a set of structurally diverse molecules that bind to KEAP1 with submicromolar affinity.</p>
<ul>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/654a339b48dad2312043870c">Deep Learning Strategies for Enhanced Molecular Docking and Virtual Screening</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41596-023-00885-w">A practical guide to machine-learning scoring for structure-based virtual screening</a></p></li>
<li><p><a href="https://wires.onlinelibrary.wiley.com/doi/10.1002/wcms.1678">Keeping pace with the explosive growth of chemical libraries with structure-based virtual screening</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41586-019-0917-9">Lyu, Jiankun, et al.&nbsp;“Ultra-large library docking for discovering new chemotypes.” Nature 566.7743 (2019): 224-229.</a></p></li>
</ul>
<p>Researchers at UCSF looking at large scale docking for making ultra-large libraries accessible. They dock 170 million make-on-demand compounds from 130 well characterized reactions. Found new chemotypes that have interaction with 2 targets.</p>
</section>
<section id="fragment-based-drug-discovery" class="level2">
<h2 class="anchored" data-anchor-id="fragment-based-drug-discovery">Fragment-based drug discovery</h2>
<p><strong>What is a fragment?</strong></p>
<p>In fragment-based drug discovery (FBDD), a fragment is a small, low-molecular-weight chemical entity typically ranging from 120-250 Daltons, with properties such as cLogP &lt; 2.5, hydrogen atom count (HAC) of 9-18, hydrogen bond acceptors (HBA) &lt; 7, and hydrogen bond donors (HBD) &lt; 4, as specified by Asinex. These fragments serve as starting points for drug development. They bind to target proteins with low affinity but high efficiency, enabling the identification of key interactions. By optimizing and linking fragments, researchers can develop potent lead compounds, enhancing the efficiency of the drug discovery process and improving hit finding.</p>
<p><strong>Known collection</strong></p>
<ul>
<li><a href="https://www.cambridgemedchemconsulting.com/resources/hit_identification/fragment_collections.html">Cambridge Chem Consulting</a></li>
<li><a href="https://www.evotec.com/en/investor-relations/news/p/evotec-enhances-its-fragment-based-drug-discovery-business-by-the-addition-of-nmr-screening-assets-from-combinature-4496">Evotec</a></li>
<li><a href="https://www.diamond.ac.uk/Instruments/Mx/Fragment-Screening/Fragment-Libraries.html">Diamond light source</a></li>
</ul>
<p><strong>Blogs</strong></p>
<ul>
<li><a href="https://practicalfragments.blogspot.com/2024/03/fragments-2024.html">Practical Fragment blog</a></li>
</ul>
<p><strong>Book Chapters</strong></p>
<ul>
<li><a href="https://doi.org/10.1038/nrd1467">Rees, D. C.; Congreve, M.; Murray, C. W.; Carr, R. Fragment-Based Lead Discovery. Nat Rev Drug Discov 2004, 3 (8), 660–672</a></li>
</ul>
<p>The paper by Rees, D. C.; Congreve, M.; Murray, C. W.; Carr, R. discusses the concept of fragment-based lead discovery in drug development. The authors highlight the challenges in the drug discovery pipeline, particularly the ‘target-rich, lead-poor’ issue and the high attrition rate of chemical compounds in preclinical development. They discuss the use of fragment-based approaches as a solution, which involves the selection, screening, and optimization of fragments (also referred to as needles, shapes, binding elements, or seed templates) for lead identification. This approach requires significantly fewer compounds to be screened and synthesized, and has a high success rate in generating chemical series with lead-like properties. The authors also discuss different strategies for developing fragments into high-affinity leads, such as fragment evolution and fragment linking. The paper includes examples from 25 different protein targets to illustrate these strategies.</p>
<p><strong>Articles</strong></p>
<ul>
<li><a href="https://chemrxiv.org/engage/chemrxiv/article-details/65d751ab9138d23161b7ea38">Fragmenstein: predicting protein-ligand structures of compounds derived from known crystallographic fragment hits using a strict conserved-binding–based methodology</a></li>
</ul>
<p>Fragmenstein, stitches ligand atoms from known fragment hits to predict protein-ligand complex conformations more accurately. Fragmenstein uses a Python package to merge or place compounds by stitching together atoms from fragment hits and then energy minimizing them under strong constraints.</p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.4c00154">Zhang, Yueqing, et al.&nbsp;“FragGrow: A Web Server for Structure-Based Drug Design by Fragment Growing within Constraints.” Journal of Chemical Information and Modeling (2024).</a></p></li>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/667699175101a2ffa837f110">Cree, Ben, et al.&nbsp;“Active learning driven prioritisation of compounds from on-demand libraries targeting the SARS-CoV-2 main protease.” (2024).</a><a href="https://github.com/cole-group/FEgrow/">FEGrow Github</a></p></li>
<li><p><a href="https://www.nature.com/articles/s42256-024-00815-9">Igashov, Ilia, et al.&nbsp;“Equivariant 3D-conditional diffusion model for molecular linker design.” Nature Machine Intelligence (2024): 1-11.</a> <a href="https://iwatobipen.wordpress.com/2024/04/24/generate-new-molecules-from-fragments-with-diffusion-model-cheminformatics-rdkit-difflinker-memo/">Pen’s blog</a></p></li>
<li><p><a href="https://pubmed.ncbi.nlm.nih.gov/35989379/">Penner, Patrick, et al.&nbsp;“FastGrow: on-the-fly growing and its application to DYRK1A.” Journal of Computer-Aided Molecular Design 36.9 (2022): 639-651.</a> <a href="https://practicalfragments.blogspot.com/2022/09/growing-fragments-in-silico-with.html">Pen’s blog</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.2c01004">Carbery, Anna, et al.&nbsp;“Fragment libraries designed to be functionally diverse recover protein binding information more efficiently than standard structurally diverse libraries.” Journal of Medicinal Chemistry 65.16 (2022): 11404-11413.</a>. <a href="https://practicalfragments.blogspot.com/2022/08/diverse-function-not-structure-in.html">Practical Fragment Blog</a></p></li>
<li><p><a href="https://doi.org/10.1126/science.abo7201">Boby, M. L. Open Science Discovery of Potent Noncovalent SARS-CoV-2 Main Protease Inhibitors. Science 2023, 382 (6671)</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.2c00813">Müller, Janis, et al.&nbsp;“Magnet for the needle in haystack:“crystal structure first” fragment hits unlock active chemical matter using targeted exploration of vast chemical spaces.” Journal of Medicinal Chemistry 65.23 (2022): 15663-15678.</a>. <a href="https://www.science.org/content/blog-post/crystal-fragments?utm_source=pocket_mylist">Blog</a></p></li>
</ul>
<p>The authors use a fragment screening approach to look at hits for protein kinase target and instead of using biophysical assay in fragment screening use crystallographic data directly to learn the conformation of the fragments. They find 4 ‘seed’ substructures which fit nicely in the protein(not affinity) and use those to inform the latter expansion which is done through the Enamine REAL dataset and known reaction classes. What I liked the most and found interesting is the high throughput binding pose and docking workflow of 200k compounds, the large scale crystallographic fragment hit analysis, and the focused curated library generation using Enamine REAL dataset. I was curious to know what seasoned experts had to say about this.</p>
<ul>
<li><a href="https://www.eyesopen.com/brood">BROOD</a></li>
</ul>
<p>Commercial software solution from OpenEye for fragment exchange</p>
</section>
<section id="protein-engineering" class="level2">
<h2 class="anchored" data-anchor-id="protein-engineering">Protein engineering</h2>
<ul>
<li><a href="https://www.nature.com/articles/s41467-023-37701-8">PeSTo: parameter-free geometric deep learning for accurate prediction of protein binding interfaces</a></li>
</ul>
</section>
<section id="protein-ligand-interactions" class="level2">
<h2 class="anchored" data-anchor-id="protein-ligand-interactions">Protein-ligand interactions</h2>
<ul>
<li><a href="https://wires.onlinelibrary.wiley.com/doi/10.1002/wcms.1711">Yim, Jason, et al.&nbsp;“Diffusion models in protein structure and docking.” Wiley Interdisciplinary Reviews: Computational Molecular Science 14.2 (2024): e1711.</a></li>
</ul>
<p>Nice review of the field that looks at computational model to predict protein-ligand interaction and molecular docking. In recent times, diffusion-based model have shown promising performance. This review documents the current state of the field and next steps. This survey covers DMs primarily from the point of view of backbone generation, both unconditional and conditional generation. It is interesting to see how modeling the backbone, sequence, and side chains would bring further benefit beyond the current strategy of modeling them one after the other.</p>
<ul>
<li><p><a href="https://arxiv.org/abs/2210.01776">DiffDock</a></p></li>
<li><p><a href="https://www.nature.com/articles/s42256-024-00792-z">Qiao, Zhuoran, Weili Nie, Arash Vahdat, Thomas F. Miller III, and Animashree Anandkumar. “State-specific protein–ligand complex structure prediction with a multiscale deep generative model.” Nature Machine Intelligence 6, no. 2 (2024): 195-208.</a></p></li>
</ul>
<p>Iambic Therapeutics’ AI model that predicts the combined shape of proteins and small molecules outperforms Google DeepMind’s AlphaFold. Lambic’s newest model, called NeuralPLexer2, had a 75% success rate in predicting these protein-ligand structures. AlphaFold’s latest version, as described last October in a blog post, was 74% successful. But Iambic’s model jumps to a 93% success rate when the model receives additional info on amino acids near the small molecule.</p>
<ul>
<li><a href="https://www.science.org/doi/10.1126/science.adl2528">RFdiffusion All-Atom</a> <a href="https://github.com/baker-laboratory/rf_diffusion_all_atom">Github</a></li>
</ul>
<p>RoseTTAFold All-Atom (RFAA) was trained to represent amino acids and DNA bases at the residue level and all other chemical groups at the atomic level, allowing it to accurately model proteins and the other chemicals they so often interact with.</p>
<p>RFdiffusion All-Atom: build bespoke protein structures around small molecules. The team designed and experimentally validated, through crystallography and binding measurements, proteins that bind the cardiac disease therapeutic digoxigenin, the enzymatic cofactor heme, and the light-harvesting molecule bilin. Although there is still room for improvement in prediction accuracy, we anticipate that RFAA should be broadly useful for modeling full biological assemblies and RFdiffusionAA for designing small molecule–binding proteins and sensors.</p>
<ul>
<li><a href="https://www.nature.com/articles/s41467-024-45461-2">DynamicBind</a></li>
</ul>
<p>DynamicBind, a deep learning method that employs equivariant geometric diffusion networks to construct a smooth energy landscape, promoting efficient transitions between different equilibrium states. DynamicBind accurately recovers ligand-specific conformations from unbound protein structures without the need for holo-structures or extensive sampling.</p>
<ul>
<li><a href="https://www.nature.com/articles/s43588-023-00529-9?#Sec13">Yu, Jie, et al.&nbsp;“Computing the relative binding affinity of ligands based on a pairwise binding comparison network.” Nature Computational Science 3.10 (2023): 860-872.</a></li>
</ul>
<section id="conformer-generators" class="level3">
<h3 class="anchored" data-anchor-id="conformer-generators">Conformer generators</h3>
<ul>
<li><p><a href="https://arxiv.org/abs/2502.13220">The impact of conformer quality on learned representations of molecular conformer ensembles</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.3c01245">McNutt, Andrew T., et al.&nbsp;“Conformer Generation for Structure-Based Drug Design: How Many and How Good?.” Journal of Chemical Information and Modeling (2023).</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.3c01519">Wang, Zhe, et al.&nbsp;“Small-Molecule Conformer Generators: Evaluation of Traditional Methods and AI Models on High-Quality Data Sets.” Journal of Chemical Information and Modeling (2023).</a></p></li>
<li><p><a href="https://arxiv.org/pdf/2310.00115.pdf">Zhu, Yanqiao, et al.&nbsp;“Learning Over Molecular Conformer Ensembles: Datasets and Benchmarks.” arXiv preprint arXiv:2310.00115 (2023).</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/abs/10.1021/acs.jcim.7b00505?src=recsys">Friedrich, Nils-Ole, et al.&nbsp;“Benchmarking commercial conformer ensemble generators.” Journal of chemical information and modeling 57.11 (2017): 2719-2728.</a></p></li>
</ul>
<p>The paper benchmarks eight commercial conformer ensemble generators: OMEGA, ConfGenX, iCon, MOE LowModeMD, MacroModel, LigPrep, OpenEye, and Schrödinger, along with RDKit. It finds that commercial algorithms are highly robust and accurate, with OMEGA being the top performer in terms of accuracy and speed. RDKit shows competitive performance compared to mid-ranked commercial algorithms. The study highlights the impact of rotatable bonds on success rates and recommends enabling built-in clustering for better accuracy.</p>
<p><strong>Structure Quality Assessment</strong></p>
<ul>
<li><a href="https://arxiv.org/abs/2308.05777">Buttenschoen, Martin, Garrett M. Morris, and Charlotte M. Deane. “PoseBusters: AI-based docking methods fail to generate physically valid poses or generalise to novel sequences.” arXiv preprint arXiv:2308.05777 (2023).</a></li>
</ul>
<p>Python package for evaluating the quality of docked poses. PoseBusters performs a series of geometry checks on docked poses and also evaluates intra and inter-molecular interactions. The authors used the Astex Diverse Set and a newly developed PoseBusters benchmark set to evaluate five popular deep learning docking programs and two conventional docking approaches. The conventional docking programs dramatically outperformed the deep learning methods on both datasets. In most cases, more than half of the solutions generated by the DL docking programs failed the PoseBusters validity tests. In contrast, with the conventional docking programs, only 2-3% of the docked poses failed to validate.</p>
<ul>
<li><a href="https://arxiv.org/abs/2405.14108">Morehead, Alex, et al.&nbsp;“Deep Learning for Protein-Ligand Docking: Are We There Yet?.” arXiv preprint arXiv:2405.14108 (2024)</a><a href="https://github.com/BioinfoMachineLearning/PoseBench">Github</a></li>
</ul>
<p>Suite of tools and workflow to benchmark Deep learning method’s ability to predict protein-ligand interaction modeling - from apo to halo configuration for P/L pairs. The authors find that all but one DL method fail to generalize to multi-ligand protein targets.</p>
<p><a href="https://arxiv.org/abs/2308.07413">Benchmarking Generated Poses: How Rational is Structure-based Drug Design with Generative Models</a></p>
<p>PoseCheck evaluates steric clashes, ligand strain energy, and intramolecular interactions to identify problematic structures. In addition, structures are redocked with AutoDock Vina to confirm the validity of the proposed binding mode. In evaluating several recently published generative models, the authors identify failure modes that will hopefully influence future work on structure-based generative design.</p>
</section>
<section id="binding-energetic-prediction" class="level3">
<h3 class="anchored" data-anchor-id="binding-energetic-prediction">Binding energetic prediction</h3>
<ul>
<li><p><a href="https://www.nature.com/articles/s42004-025-01428-y">Valsson, Ísak, et al.&nbsp;“Narrowing the gap between machine learning scoring functions and free energy perturbation using augmented data.” Communications Chemistry 8.1 (2025): 41.</a></p></li>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/6675a38d5101a2ffa8274f62">Warren, M., Deane, C., Magarkar, A., Morris, G., &amp; Biggin, P. (2024). How to make machine learning scoring functions competitive with FEP.</a></p></li>
</ul>
<p>Current SOTA model fail in out of distribution datasets implying overfitting or memorization of ligand-specific features. This paper introduces AEV-PLIG, atomic environment vector with protein-ligand interaction graphs. They propose new benchmark metrics and data augmentation strategies.A multi-head attention graph NN is trained with the node features of the P-L interaction. They report comparable performance to FEP+ on standard benchmarks.</p>
<ul>
<li><a href="https://www.nature.com/articles/s42256-024-00847-1">Koh, Huan Yee, et al.&nbsp;“Physicochemical graph neural network for learning protein–ligand interaction fingerprints from sequence data.” Nature Machine Intelligence (2024): 1-15.</a>. <a href="https://github.com/huankoh/PSICHIC">Github</a></li>
</ul>
<p>PSICHIC (PhySIcoCHemICal graph neural network), a framework incorporating physicochemical constraints to decode interaction fingerprints directly from sequence data alone.</p>
<ul>
<li><p><a href="https://www.nature.com/articles/s41598-023-45086-3">Janela, Tiago, and Jürgen Bajorath. “Rationalizing general limitations in assessing and comparing methods for compound potency prediction.” Scientific Reports 13.1 (2023): 17816.</a></p></li>
<li><p><a href="https://www.nature.com/articles/s43588-023-00531-1">Efficient prediction of relative ligand binding affinity in drug discovery. Nat Comput Sci 3, 829–830 (2023). https://doi.org/10.1038/s43588-023-00531-1</a></p></li>
<li><p><a href="https://link.springer.com/article/10.1007/s10822-022-00494-x">Xu, Huafeng. “The slow but steady rise of binding free energy calculations in drug discovery.” Journal of Computer-Aided Molecular Design (2022): 1-8.</a></p></li>
<li><p><a href="https://www.sciencedirect.com/science/article/pii/S2667318522000204">Thompson, James, et al.&nbsp;“Optimizing active learning for free energy calculations.” Artificial Intelligence in the Life Sciences 2 (2022): 100050.</a></p></li>
</ul>
</section>
</section>
<section id="molecular-dynamics" class="level2">
<h2 class="anchored" data-anchor-id="molecular-dynamics">Molecular-dynamics</h2>
<ul>
<li><a href="https://mdance.readthedocs.io/en/latest/about.html">MDANCE</a></li>
</ul>
<p>Molecular Dynamics Analysis with N-ary Clustering Ensembles (MDANCE) is a flexible n-ary clustering package that provides a set of tools for clustering Molecular Dynamics trajectories.</p>
</section>
<section id="cheminformatics-focus" class="level2">
<h2 class="anchored" data-anchor-id="cheminformatics-focus">Cheminformatics-focus</h2>
<p>Catalog of recent reviews and manuscripts I have found useful when learning more about the state-of-the-art in Cheminformatics. I’ve tried to categorize them roughly based on their area of application:</p>
<section id="representation" class="level3">
<h3 class="anchored" data-anchor-id="representation">Representation</h3>
<p>Small molecules to be understood by computers and used for model training have to represented in a form amenable for optimization. In addition, this form of abstraction much capture appropriate level of chemical properties so as to imbue the data-driven models with necessary chemistry and physics for modeling. A lot of times different properties of the molecules are ‘lost in translation’ or obfuscated when converting them into machine-ready forms. Formerly the process of converting molecules from one form to another is called featurization. There are different forms, methods, theories to encode the molecules. Broadly there are as follows: * Fingerprints * Descriptors * Pharmacophores * Graph-based * Natural language-based * Shape-based</p>
<p><strong>Reviews</strong></p>
<ul>
<li><p><a href="https://academic.oup.com/bib/article/25/1/bbad422/7455245">From intuition to AI: evolution of small molecule representations in drug discovery</a></p></li>
<li><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00460-5">Representation of Molecules in NN: Molecular representation in AI-driven drug discovery: review and guide</a></p></li>
</ul>
<p><strong>Articles</strong></p>
<ul>
<li><a href="https://chemrxiv.org/engage/chemrxiv/article-details/6525acea45aaa5fdbbb87ac2?utm_source=pocket_saves">Boulougouri, Maria, Pierre Vandergheynst, and Daniel Probst. “Molecular set representation learning.” (2023).</a></li>
</ul>
<p>The authors propose a new way to represent molecules, not as chemical bonds, but rather set representations. They show the set representation scheme can be alternative to SOTA graph-models and performs at par to the predictive tasks such as reaction yields and protein-ligand affinities.</p>
<ul>
<li><p><a href="https://arxiv.org/abs/1905.13741">M. Krenn, F. Hase, A. Nigam, P. Friederich, and A. Aspuru-Guzik, “Self-Referencing Embedded Strings (SELFIES): A 100% robust molecular string representation,” Mach. Learn. Sci. Technol., pp.&nbsp;1–9, 2020</a></p></li>
<li><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00479-8">Could graph neural networks learn better molecular representation for drug discovery? A comparison study of descriptor-based and graph-based models</a></p></li>
</ul>
<p>Comparative study of descriptor-based and graph-based models using public data set. Used descriptor-based models (XGBoost, RF, SVM, using ECFP) and compared them to graph-based models (GCN, GAT, AttentiveFP, MPNN). They show descriptor-based models outperform the graph-based models in terms of prediction accuracy and computational efficiency with SVM having best predictions. Graph-based methods are good for multi-task learning.</p>
<ul>
<li><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00497-0">He, Jiazhen, et al.&nbsp;“Molecular optimization by capturing chemist’s intuition using deep neural networks.” Journal of cheminformatics 13.1 (2021): 1-17.</a></li>
</ul>
<p><strong>Descriptor generation</strong></p>
<ul>
<li><p><a href="https://github.com/digital-chemistry-laboratory/morfeus">Morfeus</a> Machine learning focused descriptors for small molecules with emphasis on 3D information.</p></li>
<li><p><a href="https://github.com/mordred-descriptor/mordred">Mordred</a></p></li>
<li><p><a href="https://molfeat.datamol.io/featurizers">MolFeat</a></p></li>
<li><p><a href="https://deepchem.readthedocs.io/en/latest/api_reference/featurizers.html">DeepChem</a></p></li>
<li><p><a href="https://github.com/DrrDom/pmapper">PMapper</a></p></li>
</ul>
<p>Pmapper is a Python module to generate 3D pharmacophore signatures and fingerprints. Signatures uniquely encode 3D pharmacophores with hashes suitable for fast identification of identical pharmacophores.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acscatal.8b04043">SteriMol</a></li>
</ul>
<p>Generate conformationally sampled descriptors for a molecule. This workflow provides Boltzmann-averaged Sterimol parameters. These descriptors might be useful for problems where spatial orientation inform the selectivity or properties being trained for. <a href="https://github.com/bobbypaton/wSterimol">Github</a></p>
</section>
<section id="predictive-modeling" class="level3">
<h3 class="anchored" data-anchor-id="predictive-modeling">Predictive modeling</h3>
<p><strong>Reviews</strong></p>
<ul>
<li><a href="https://wires.onlinelibrary.wiley.com/doi/10.1002/wcms.1698">Chemical complexity challenge: Is multi-instance machine learning a solution</a></li>
</ul>
<p>Traditional ML uses the relationship between a single instance (a chemical structure) and a single label (a property). It doesn’t provide a facility for mapping multiple instances (an ensemble of conformers) to a label. There has recently been renewed interest in multiple instance learning (MIL), a technique developed over 30 years ago. MIL provides a framework that enables the mapping of conformational ensembles to properties. A recent review by Zankov from Hokkaido University and coworkers at other institutions provides an excellent overview of the challenges and opportunities associated with MIL in QSAR, genomics, and several other areas. The paper also provides links to several software packages for building MIL models.</p>
<ul>
<li><a href="https://arxiv.org/abs/2309.17161">Current Methods for Drug Property Prediction in the Real World</a></li>
</ul>
<p>Overview of the field and some factors that complicate current benchmarking efforts. The authors compared several molecular representations and ML algorithms in evaluating model accuracy and uncertainty. These evaluations highlighted the strengths of different QSAR modeling and ADME prediction methods. Consistent with other papers published in 2023, 2D descriptors performed best for ADME prediction, while Gaussian Process Regression with fingerprints was the method of choice when predicting biological activity.</p>
<ul>
<li><a href="https://www.nature.com/articles/s41598-023-45086-3">Rationalizing general limitations in assessing and comparing methods for compound potency prediction</a></li>
</ul>
<p>A paper by Janela and Bajorath outlines several limitations in current benchmarking strategies. The authors used sound statistical methodologies to examine the impact of compound potency value distributions on performance metrics associated with regression models. They found that across several different ML algorithms, there was a consistent relationship between model performance and the activity range of the dataset. These findings enabled the authors to define bounds for prediction accuracy. The method used in this paper should be informative to those designing future benchmarks.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/abs/10.1021/acs.jcim.9b00237">Yang, K., Swanson, K., Jin, W., Coley, C., Eiden, P., Gao, H., Guzman-Perez, A., Hopper, T., Kelley, B., Mathea, M. and Palmer, A., 2019. Analyzing learned molecular representations for property prediction. Journal of chemical information and modeling, 59(8), pp.3370-3388</a></li>
</ul>
<p>Benchmark property prediction models on 19 public and 16 proprietary industrial data sets spanning a wide variety of chemical end points. Introduce a modeling framework (<strong>Chemprop</strong>) that consistently matches or outperforms models using fixed molecular descriptors as well as previous graph neural architectures on both public and proprietary data sets.</p>
<ul>
<li><a href="https://arxiv.org/abs/2107.10402">Stuyver, T. and Coley, C.W., 2021. Quantum chemistry-augmented neural networks for reactivity prediction: Performance, generalizability and interpretability. arXiv preprint arXiv:2107.10402</a></li>
</ul>
<p>Combine structure (Graph-networks) and descriptor based features (QM-derived) to predict activation energies (E<sub>2</sub>/SN<sub>2</sub> barrier height prediction) and regioselectivity. Incorporating QM and structure leads to better overall accuracy and generalizability even in low data regions. Atom and bond level features derived using QM and used in the model generation with a smaller dataset.</p>
<p><strong>Transformer-based</strong></p>
<ul>
<li><a href="https://arxiv.org/pdf/2406.12056v3">InfoAlign</a></li>
<li><a href="https://arxiv.org/pdf/2010.09885">ChemBERTa</a></li>
</ul>
<p><strong>pKa prediction</strong></p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jctc.4c00328">Abarbanel, Omri, and Geoffrey Hutchison. “QupKake: Integrating Machine Learning and Quantum Chemistry for micro-pKa Predictions.” (2023).</a>. <a href="https://github.com/hutchisonlab/QupKake">Github</a></li>
</ul>
<p>QupKake combines GFN2-xTB calculations with graph-neural-networks to accurately predict micro-pKa values of organic molecules.</p>
<p><strong>Hydrogen bond energy</strong></p>
<ul>
<li><a href="https://www.nature.com/articles/s41598-023-30089-x">Jazzy: Fast calculation of hydrogen-bond strengths and free energy of hydration of small molecules</a></li>
</ul>
</section>
</section>
<section id="pharmacophore-shape-searching" class="level2">
<h2 class="anchored" data-anchor-id="pharmacophore-shape-searching">Pharmacophore / shape searching</h2>
<ul>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.4c00453">Rianjongdee, Francesco, et al.&nbsp;“bbSelect–An Open-Source Tool for Performing a 3D Pharmacophore-Driven Diverse Selection of R-Groups.” Journal of Chemical Information and Modeling (2024).</a>. <a href="https://iwatobipen.wordpress.com/2024/06/25/new-approach-for-selecting-builidng-blocks-from-reagent-set-rdkit-memo-cheminformatics/">Pen’s blog</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.4c00516">PheSA: An Open-Source Tool for Pharmacophore-Enhanced Shape Alignment</a></p></li>
</ul>
<p>PheSA is a CPU bound algorithmic improvement over ROCS shape/color alignment that gives you an option to incorporate binding site knowledge.</p>
<ul>
<li><p><a href="https://users.abo.fi/mivainio/shaep/download.php">ShaEP</a></p></li>
<li><p><a href="https://github.com/molecularinformatics/roshambo">Roshambo</a></p></li>
</ul>
<p>ROSHAMBO is a GPU accelerated implementation of ROCS</p>
<section id="qsar-benchmarks" class="level3">
<h3 class="anchored" data-anchor-id="qsar-benchmarks">QSAR benchmarks</h3>
<ul>
<li><a href="https://www.nature.com/articles/s41597-024-03105-6">Llompart, P., et al.&nbsp;“Will we ever be able to accurately predict solubility?” Scientific Data 11.1 (2024): 303.</a></li>
</ul>
<p>The paper discusses challenges in predicting thermodynamic solubility with machine learning. It reviews historical data, analyzes solubility datasets, and assesses model reliability. The authors propose a workflow for data curation and present new models and quality-assessed datasets for public use.</p>
<ul>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/659d3878e9ebbb4db9c3088f">Multi-task ADME/PK Prediction at Industrial Scale: Leveraging Large and Diverse Experimental Datasets</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41467-023-41948-6">Deng, Jianyuan, et al.&nbsp;“A systematic study of key elements underlying molecular property prediction.” Nature Communications 14.1 (2023): 6395.</a></p></li>
</ul>
<p>Deng and coworkers from Stony Brook University compared many popular ML algorithms and representations, curated new datasets, and performed statistical analysis on the results. This paper provides one of the best comparisons of ML methods published to date. The authors compare fixed representations, such as molecular fingerprints, with representations learned from SMILES strings and molecular graphs and conclude that, in most cases, the fixed representations provide the best performance. Another interesting aspect of this paper was an attempt to establish a relationship between dataset size and the performance of different molecular representations. While fixed representations performed well on smaller datasets, learned representations didn’t become competitive until between 6K and 100K datapoints were available.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.3c00160">Fang, Cheng, et al.&nbsp;“Prospective Validation of Machine Learning Algorithms for Absorption, Distribution, Metabolism, and Excretion Prediction: An Industrial Perspective.” Journal of Chemical Information and Modeling (2023).</a></li>
</ul>
<p>A paper from Fang and coworkers at Biogen introduced several new ADME datasets. Unlike most literature benchmarks, which contain data collected from dozens of papers, these experiments were consistently performed by the same people in the same lab. The authors provided prospective comparisons of several widely used ML methods, including random forest, SVM, XGBoost, LightGBM, and message-passing neural networks (MPNNs) on several relevant endpoints, including aqueous solubility, metabolic stability, membrane permeability, and plasma protein binding.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.3c01083">Beckers, Maximilian, et al.&nbsp;“Prediction of Small-Molecule Developability Using Large-Scale In Silico ADMET Models.” Journal of Medicinal Chemistry (2023).</a></li>
</ul>
<p>The paper presents a novel deep learning approach to predict the developability of small molecules based on their predicted ADMET properties. The authors use a large-scale data set of compounds from the Novartis pipeline and train a neural network to rank compounds according to their potential to progress beyond in vivo PK studies. The resulting score, called bPK, outperforms other compound scoring methods and shows strong generalization ability on public data. The authors demonstrate the usefulness of bPK for series prioritization and optimization in drug discovery projects.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.2c01073">D van Tilborg, Derek, Alisa Alenicheva, and Francesca Grisoni. “Exposing the limitations of molecular machine learning with activity cliffs.” Journal of Chemical Information and Modeling 62.23 (2022): 5938-5951.</a></li>
</ul>
<p>Account on how to treat and analyze activity cliffs in context of developing a predictive model. The authors outline best practices to probe activity cliffs. They show, using 24 DL and ML models and 30 targets, ML approaches based on molecular descriptors outperformed more complex deep learning methods. Activity cliff pairs were defined on similarity of the molecule SMILES and the bioactivity difference. Compared to most traditional machine learning approaches, deep neural networks seem to fall short at picking up subtle structural differences (and the corresponding property change) that give rise to activity cliffs.</p>
<ul>
<li><p><a href="https://pubs.rsc.org/en/content/articlelanding/2018/sc/c8sc00148k">Large-scale comparison of machine learning methods for drug target prediction on ChEMBL - Chemical Science (RSC Publishing)</a></p></li>
<li><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-017-0232-0">Beyond the hype: deep neural networks outperform established methods using a ChEMBL bioactivity benchmark set, Journal of Cheminformatics</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.molpharmaceut.2c00962">Systematic Evaluation of Local and Global Machine Learning Models for the Prediction of ADME Properties</a></p></li>
</ul>
<p>Authors provide an evaluation of global and local models for ADME endpoint prediction. They compare the performance of global models and domain-specific local models. 10 different asays and 112 drug discovery projects were analyzed. The results showed consistent superior performance of global ADME models for property prediction. Performance improvement of global models over project-wise local models ranged from 3% to 25% in MAE. Local model improvements higher than 20% were achieved for only 7% of the assay-project pairs.</p>
<ul>
<li><a href="https://www.biorxiv.org/content/10.1101/2023.12.28.573531v1.abstract">Swanson, Kyle, et al.&nbsp;“ADMET-AI: A machine learning ADMET platform for evaluation of large-scale chemical libraries.” BioRxiv (2023): 2023-12.</a> <a href="https://admet.ai.greenstonebio.com/">Web interface</a> <a href="https://github.com/swansonk14/admet_ai">Code</a></li>
</ul>
<p>Online web interface to quickly predict ADMET properties using specific AI models trained on Therapeutic Data Commmons dataset.</p>
<p><strong>Datasets</strong></p>
<ul>
<li><a href="https://www.nature.com/articles/s41597-022-01390-7">QMugs</a></li>
</ul>
<p>QMugs (Quantum mechanical properties of drug-like molecules) collection comprises quantum mechanical properties of more than 665 k biologically and pharmacologically relevant molecules extracted from the ChEMBL database, totaling 2M conformers.</p>
<ul>
<li><p><a href="https://pubs.rsc.org/en/content/articlelanding/2018/sc/c7sc02664a">MoleculeNet: a benchmark for molecular machine learning (rsc.org)</a></p></li>
<li><p><a href="https://tdcommons.ai/">TDC: Therapeutic Data Commons</a></p></li>
</ul>
</section>
<section id="matched-molecular-pair" class="level3">
<h3 class="anchored" data-anchor-id="matched-molecular-pair">Matched molecular-pair</h3>
<ul>
<li><p><a href="https://link.springer.com/article/10.1023/A:1021271615909">Raymond, John W., and Peter Willett. “Maximum common subgraph isomorphism algorithms for the matching of chemical structures.” Journal of computer-aided molecular design 16.7 (2002): 521-533.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.8b00173">Dalke, Andrew, Jerome Hert, and Christian Kramer. “mmpdb: An open-source matched molecular pair platform for large multiproperty data sets.” Journal of chemical information and modeling 58.5 (2018): 902-910.</a></p></li>
</ul>
</section>
<section id="r-group-replacement-dataset" class="level3">
<h3 class="anchored" data-anchor-id="r-group-replacement-dataset">R-group replacement dataset</h3>
</section>
<section id="enumeration-of-chemical-space" class="level3">
<h3 class="anchored" data-anchor-id="enumeration-of-chemical-space">Enumeration of chemical space</h3>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.1c01378">Bellmann, Louis, et al.&nbsp;“Comparison of Combinatorial Fragment Spaces and Its Application to Ultralarge Make-on-Demand Compound Catalogs.” Journal of Chemical Information and Modeling (2022).</a></li>
</ul>
<p>Authors propose an algorithmic approach called as SpaceCompare to calculate overlap and diversity of the ultra-large combinatorial chemical libraries. The tool uses topological fragment spaces to capture the subtlties of the reaction having same product but different reactant substructures.</p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.6b00173">Nicolaou, Christos A., et al.&nbsp;“The proximal lilly collection: Mapping, exploring and exploiting feasible chemical space.” Journal of chemical information and modeling 56.7 (2016): 1253-1266.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.1c00754">Zabolotna, Y., et al.&nbsp;(2021). “SynthI: A New Open-Source Tool for Synthon-Based Library Design.” Journal of Chemical Information and Modeling.</a></p></li>
</ul>
<p>Interesting work on de-novo design of molecules wherein, the molecules being created are made up from the fragments that is known to exist and are available to the user. New molecules are generated based on the fragmented (synthons) made available in the dataset.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.1c01041">Fully Automated Creation of Virtual Chemical Fragment Spaces Using the Open-Source Library OpenChemLib</a></li>
</ul>
<p>Open-source tool to generate synthetically accessible chemical spaces using reaction definitions and building blocks. Virtual fragments are generated using one-step reaction and real-world building blocks - the workflow also support 2-3 steps creation.</p>
<ul>
<li><a href="https://onlinelibrary.wiley.com/doi/10.1002/minf.202100068">Zabolotna, Yuliana, et al.&nbsp;“NP navigator: a new look at the natural product chemical space.” Molecular informatics 40.9 (2021): 2100068.</a>.</li>
</ul>
<p>Organizing the chemical space of ChEMBL, and ZINC to compare its overlap with natural products through COCONUT. Generative Topological Mapping is used for the clustering and analysis. Helpful overview of the method with its application to drug discovery can be found <a href="https://www.sciencedirect.com/science/article/pii/S1740674920300044">here</a></p>
</section>
</section>
<section id="explainableinterpretable-machine-learning" class="level2">
<h2 class="anchored" data-anchor-id="explainableinterpretable-machine-learning">Explainable/Interpretable Machine Learning</h2>
<p><strong>Reviews/Perspectives</strong></p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jctc.2c01235">Wellawatte, Geemi P., et al.&nbsp;“A Perspective on Explanations of Molecular Prediction Models.” (2022).</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jmedchem.1c01789">Rodríguez-Pérez, Raquel, and Jürgen Bajorath. “Explainable Machine Learning for Property Predictions in Compound Optimization.” Journal of medicinal chemistry 64.24 (2021): 17744-17752</a></p></li>
</ul>
<p><strong>Articles</strong></p>
<ul>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/613268f0d5f0803706ba0c79">Wellawatte, Geemi P., Aditi Seshadri, and Andrew D. White. “Model agnostic generation of counterfactual explanations for molecules.” (2021).</a></p></li>
<li><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00519-x">Matveieva, Mariia, and Pavel Polishchuk. “Benchmarks for interpretation of QSAR models.” Journal of cheminformatics 13.1 (2021): 1-20</a>. <a href="https://patwalters.github.io/practicalcheminformatics/jupyter/ml/interpretability/2021/06/03/interpretable.html">Patrick Walter’s blog</a></p></li>
</ul>
</section>
<section id="uncertainty-quantification" class="level2">
<h2 class="anchored" data-anchor-id="uncertainty-quantification">Uncertainty quantification</h2>
<ul>
<li><p><a href="https://proceedings.neurips.cc/paper_files/paper/2023/file/f2b1b2e974fa5ea622dd87f22815f423-Paper-Conference.pdf">GAUCHE: A Library for Gaussian Processes in Chemistry</a></p></li>
<li><p><a href="https://www.sciencedirect.com/science/article/pii/S1359644620305110?via%3Dihub">Mervin, L. H., Johansson, S., Semenova, E., Giblin, K. A., &amp; Engkvist, O. (2021). Uncertainty quantification in drug design. Drug discovery today, 26(2), 474-489.</a></p></li>
<li><p><a href="https://arxiv.org/pdf/2102.11439.pdf">Alan Aspuru-Guzik perspective on uncertainty and confidence</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/abs/10.1021/acs.jcim.0c00502">Uncertainty Quantification Using Neural Networks for Molecular Property Prediction. J. Chem. Inf. Model. (2020) Hirschfeld, L., Swanson, K., Yang, K., Barzilay, R. &amp; Coley, C. W.</a></p></li>
</ul>
<p>Benchmark different models and uncertainty metrics for molecular property prediction.</p>
<ul>
<li><a href="https://arxiv.org/abs/1910.02600">Evidential Deep learning for guided molecular property prediction and disocovery Ava Soleimany, Conor Coley, et. al.</a>. <a href="https://slideslive.com/38942396/evidential-deep-learning-for-guided-molecular-property-prediction-and-discovery">Slides</a></li>
</ul>
<p>Train network to output the parameters of an evidential distribution. One forward-pass to find the uncertainty as opposed to dropout or ensemble - principled incorporation of uncertainties</p>
<ul>
<li><p><a href="https://arxiv.org/pdf/2101.11588.pdf">Differentiable sampling of molecular geometries with uncertainty-based adversarial attacks</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acscentsci.0c00026">J. P. Janet, S. Ramesh, C. Duan, H. J. Kulik, ACS Cent. Sci. 2020</a></p></li>
</ul>
<p>Conduct a global multi-objective optimization with expected improvement criterion. Find transition metal complex redox couples for Redox flow batteries that address stability, solubility, and redox potential metric. Use distance of a point from a training data in latent space as a metric to quantify uncertainty.</p>
</section>
<section id="active-learning" class="level2">
<h2 class="anchored" data-anchor-id="active-learning">Active Learning</h2>
<p>Active learning provides strategies for efficient screening of subsets of the library. In many cases, we can identify a large portion of the most promising molecules with a fraction of the compute cost.</p>
<p><strong>Blogs</strong></p>
<ul>
<li><a href="https://variationalai.substack.com/p/how-to-drug-a-novel-target-in-500?r=p9ku5&amp;utm_campaign=post&amp;utm_medium=web&amp;triedRedirect=true">How to drug a novel target in 500 molecules</a></li>
</ul>
<p><strong>Comparisons</strong></p>
<p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/655f22eecf8b3c3cd7ef43d7">Traversing Chemical Space with Active Deep Learning</a></p>
<p>The authors compared six active learning approaches on three benchmark datasets and concluded that the acquisition function is critical to AL performance. When comparing molecular representations, they found that fingerprints generalized better than graph neural networks. Consistent with previous studies, they found that the choice of an initial training set had little impact on the outcome of an AL model.</p>
<p><strong>Articles</strong></p>
<ul>
<li><a href="https://pubs.rsc.org/en/content/articlelanding/2024/sc/d3sc04653b">Dodds, M., Guo, J., Löhr, T., Tibo, A., Engkvist, O., &amp; Janet, J. P. (2024). Sample efficient reinforcement learning with active learning for molecular design. Chemical Science, 15(11), 4146-4160.</a></li>
</ul>
<p>Active learning system linked with Reinforcement learning model for molecule design through REINVENT-Active Learning. They aim to improve the sampling efficiency of the generation required to improving the MPO. Compounds discovered through RL-AL approach show significant enrichment in the MPO score.</p>
<ul>
<li><a href="https://www.biorxiv.org/content/10.1101/2024.08.25.609621v1">Correy, Galen J., Moira M. Rachman, Takaya Togo, Stefan Gahbauer, Yagmur U. Doruk, Maisie GV Stevens, Priyadarshini Jaishankar et al.&nbsp;“Extensive exploration of structure activity relationships for the SARS-CoV-2 macrodomain from shape-based fragment merging and active learning.” bioRxiv (2024): 2024-08.</a></li>
</ul>
<p>Very nice work from Shoichet (Zinc, UCSF) + Relay folks (Pat Walter et al) expanding on their active learning method to SAR CoV2. Use FrankenROCS and Thompson sampling to screen millions of compounds, identifying hits with IC50 values as low as 130 μM + ~100 X-ray crystal structures with binding data. Have been a big fan of active learning esp.&nbsp;multi-armed bandits.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.2c01052">Gusev, Filipp, et al.&nbsp;“Active learning guided drug design lead optimization based on relative binding free energy modeling.” Journal of Chemical Information and Modeling 63.2 (2023): 583-594.</a></li>
</ul>
<p>Active learning on BDE.</p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.3c01790">Klarich, Kathryn, et al.&nbsp;“Thompson Sampling─ An Efficient Method for Searching Ultralarge Synthesis on Demand Databases.” Journal of Chemical Information and Modeling (2024).</a>. Update on it from AZ folks <a href="https://www.biorxiv.org/content/10.1101/2024.05.16.594622v1.full.pdf">Link</a></p></li>
<li><p><a href="https://pubs.rsc.org/en/content/articlelanding/2024/dd/d3dd00227f">Fromer, Jenna C., David E. Graff, and Connor W. Coley. “Pareto optimization to accelerate multi-objective virtual screening.” Digital Discovery (2024).</a></p></li>
<li><p><a href="https://pubs.rsc.org/en/content/articlehtml/2021/sc/d0sc06805e">Graff, David E., Eugene I. Shakhnovich, and Connor W. Coley. “Accelerating high-throughput virtual screening through molecular pool-based active learning.” Chemical science 12.22 (2021): 7866-7881.</a>. <a href="https://github.com/coleygroup/molpal">GitHub</a></p></li>
</ul>
<p>Article talks about MolPAL as an active learning methodology. The team explores the application of these techniques to computational docking datasets and assess the impact of surrogate model architecture, acquisition function, and acquisition batch size on optimization performance. We observe significant reductions in computational costs; for example, using a directedmessage passing neural network we can identify 94.8% or 89.3% of the top-50 000 ligands in a 100M member library after testing only 2.4% of candidate ligands using an upper confidence bound or greedy acquisition strategy, respectively.</p>
<ul>
<li><a href="https://www.sciencedirect.com/science/article/pii/S2667318522000204">Thompson, James, et al.&nbsp;“Optimizing active learning for free energy calculations.” Artificial Intelligence in the Life Sciences 2 (2022): 100050.</a></li>
</ul>
<p>Article exploring different active learning strategies for looking at sampling the congeneric RBFE calculations. The paper explores the impact of several AL design choices. They show that in their case, the overall AL performance is largely insensitive to the specific ML method and acquisition functions used. The significant factor affecting the performance was the number of molecules sampled at each iteration.</p>
<ul>
<li><p><a href="https://doi.org/10.1016/j.ddtec.2020.06.001">Reker, D. Practical Considerations for Active Machine Learning in Drug Discovery. Drug Discov. Today Technol. 2020</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acscentsci.1c00546">A. P. Soleimany, A. Amini, S. Goldman, D. Rus, S. N. Bhatia, and C. W. Coley, “Evidential Deep Learning for Guided Molecular Property Prediction and Discovery,” ACS Cent. Sci., Jul.&nbsp;2021.</a>. <a href="https://slideslive.com/38942396/evidential-deep-learning-for-guided-molecular-property-prediction-and-discovery">Slideshare</a></p></li>
</ul>
<p>Train property prediction model to output a distribution statistics in single pass that describes the uncertainty. This is in contrast to using ensemble models like MC dropout. Interesting way to estimate the epistemic (due to / from model) uncertainty in the prediction. Use this approach on antibiotic search problem of Stokes et. al.&nbsp;Compare Chemprop and SchNet models on different tasks.</p>
<p><strong>Self-contained software</strong></p>
<ul>
<li><p><a href="https://pyepal.readthedocs.io/en/latest/">Pyepal</a></p></li>
<li><p><a href="https://github.com/emdgroup/baybe">BayBE</a></p></li>
<li><p><a href="https://github.com/aspuru-guzik-group/gryffin">Gryffin</a></p></li>
<li><p><a href="https://github.com/aspuru-guzik-group/atlas">Atlas</a></p></li>
<li><p><a href="https://github.com/doyle-lab-ucla/edboplus">EDBO+</a></p></li>
<li><p>Baird, Sterling G., Andrew R. Falkowski, and Taylor D. Sparks. “Honegumi: An Interface for Accelerating the Adoption of Bayesian Optimization in the Experimental Sciences.”&nbsp;<em>arXiv</em>, February 4, 2025.&nbsp;<a href="https://doi.org/10.48550/arXiv.2502.06815">https://doi.org/10.48550/arXiv.2502.06815</a>.</p></li>
</ul>
<p><strong>SaaS companies</strong></p>
<ul>
<li><p><a href="https://www.yonedalabs.com/">Yoneda Labs</a></p></li>
<li><p><a href="https://www.react-wise.com/team">ReactWise</a></p></li>
<li><p><a href="https://allchemy.net/">Allchemy</a></p></li>
<li><p><a href="https://atinary.com/solutions/">Atinary</a></p></li>
<li><p><a href="https://www.chemify.io/">Chemify</a></p></li>
</ul>
<section id="multi-parameter-optimization" class="level3">
<h3 class="anchored" data-anchor-id="multi-parameter-optimization">Multi-parameter optimization</h3>
<ul>
<li><p><a href="https://www.cell.com/patterns/fulltext/S2666-3899(23)00001-6">Computer-aided multi-objective optimization in small molecule discovery</a></p></li>
<li><p><a href="https://arxiv.org/abs/2310.10598">Pareto Optimization to Accelerate Multi-Objective Virtual Screening</a></p></li>
</ul>
</section>
</section>
<section id="transfer-learning" class="level2">
<h2 class="anchored" data-anchor-id="transfer-learning">Transfer Learning</h2>
<p><strong>Reviews</strong></p>
<ul>
<li><a href="https://pubs.acs.org/doi/pdf/10.1021/acs.jmedchem.9b02147">Cai, Chenjing, et al.&nbsp;“Transfer learning for drug discovery.” Journal of Medicinal Chemistry 63.16 (2020): 8683-8694.</a></li>
</ul>
<p><strong>Articles</strong></p>
<ul>
<li><a href="https://www.nature.com/articles/s41467-019-10827-4">Approaching coupled cluster accuracy with a general-purpose neural network potential through transfer learning</a></li>
</ul>
<p>Transfer learning by training a network to DFT data and then retrain on a dataset of gold standard QM calculations (CCSD(T)/CBS) that optimally spans chemical space. The resulting potential is broadly applicable to materials science, biology, and chemistry, and billions of times faster than CCSD(T)/CBS calculations.</p>
<ul>
<li><a href="https://iopscience.iop.org/article/10.1088/2632-2153/abae75/meta">Improving the generative performance of chemical autoencoders through transfer learning</a></li>
</ul>
</section>
<section id="meta-learning" class="level2">
<h2 class="anchored" data-anchor-id="meta-learning">Meta Learning</h2>
<ul>
<li><a href="https://pubs.acs.org/doi/abs/10.1021/acscentsci.6b00367">Altae-Tran, H., Ramsundar, B., Pappu, A. S., &amp; Pande, V. (2017). Low data drug discovery with one-shot learning. ACS central science, 3(4), 283-293.</a></li>
</ul>
<p>Authors demonstrate how one-shot learning can be used to signifinicantly lower the amount of data required to make predictions in drug discovery tasks. LSTM combined with GCNNs is shown to improve learning capabilities of the model. In the simplest one-shot learning formalism these continuous vectors are then fed into a simple nearest-neighbor classifier that labels new examples by distance-weighted combination of support set labels</p>
<ul>
<li><a href="https://arxiv.org/pdf/2003.05996.pdf">Nguyen, C. Q., Kreatsoulas, C., &amp; Branson, K. M. (2020). Meta-learning GNN initializations for low-resource molecular property prediction. arXiv preprint arXiv:2003.05996.</a></li>
</ul>
<p>Use CheMBL dataset to train a gated graph neural network (GGNN) for prediction and classification tasks using meta learning protocols. Show appreciable model performance even with just approx. 256 datapoints.</p>
</section>
<section id="federated-learning" class="level2">
<h2 class="anchored" data-anchor-id="federated-learning">Federated Learning</h2>
<ul>
<li><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00576-2">Simm, Jaak, et al.&nbsp;“Splitting chemical structure data sets for federated privacy-preserving machine learning.” Journal of Cheminformatics 13.1 (2021): 1-14.</a></p></li>
<li><p><a href="https://www.melloddy.eu/objectives">Melloddy consortium</a></p></li>
</ul>
<p>Consortia comprising of leading resarch labs and companies working on decentralized datasets and predictive modeling of biochemical and cellular activity.</p>
</section>
<section id="generative-design" class="level2">
<h2 class="anchored" data-anchor-id="generative-design">Generative design</h2>
<p><strong>Reviews</strong></p>
<ul>
<li><p><a href="https://www.nature.com/articles/s42256-024-00843-5">Du, Y., Jamasb, A. R., Guo, J., Fu, T., Harris, C., Wang, Y., … &amp; Blundell, T. L. (2024). Machine learning-aided generative molecular design.&nbsp;<em>Nature Machine Intelligence</em>, 1-16.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.4c00247?ref=recommended">Tang, Yidan, Rocco Moretti, and Jens Meiler. “Recent Advances in Automated Structure-Based De Novo Drug Design.” Journal of Chemical Information and Modeling (2024).</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/jacs.2c13467">Anstine, Dylan M., and Olexandr Isayev. “Generative Models as an Emerging Paradigm in the Chemical Sciences.” Journal of the American Chemical Society 145.16 (2023): 8736-8750.</a></p></li>
<li><p><a href="https://pubmed.ncbi.nlm.nih.gov/33562347/">Mouchlis VD, Afantitis A, Serra A, et al.&nbsp;Advances in de Novo Drug Design: From Conventional to Machine Learning Methods. Int J Mol Sci. 2021;22(4):1676. Published 2021 Feb 7. doi:10.3390/ijms22041676</a></p></li>
<li><p><a href="https://science.sciencemag.org/content/361/6400/360">B. Sanchez-Lengeling and A. Aspuru-Guzik, “Inverse molecular design using machine learning: Generative models for matter engineering,” Science (80)., vol.&nbsp;361, no. 6400, pp.&nbsp;360–365, Jul.&nbsp;2018</a></p></li>
</ul>
<p><strong>Benchmarks</strong></p>
<ul>
<li><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00861-w">Thomas, Morgan, et al.&nbsp;“MolScore: a scoring, evaluation and benchmarking framework for generative models in de novo drug design.” Journal of Cheminformatics 16.1 (2024): 1-20.</a>. <a href="https://github.com/MorganCThomas/MolScore">Github</a></li>
</ul>
<p>MolScore contains code to score and benchmark de novo compounds in the context of generative de novo design by generative models via the subpackage <code>molscore</code>, as well as, facilitate downstream evaluation via the subpackage <code>moleval</code>. An objective is defined via a JSON file which can be shared to propose new benchmark objectives, or to conduct multi-parameter objectives for drug design.</p>
<ul>
<li><a href="https://arxiv.org/abs/2112.03041">Flam-Shepherd, Daniel, Kevin Zhu, and Alán Aspuru-Guzik. “Keeping it Simple: Language Models can learn Complex Molecular Distributions.” arXiv preprint arXiv:2112.03041 (2021).</a>. <a href="https://www.nature.com/articles/s41467-022-30839-x">Nature Comms Link</a></li>
</ul>
<p>Test SOTA language models and representation performance against graph-based methods (CGVAE, JTVAE) for ‘challenging’ generative modeling tasks - generate a molecule - property distribution as a function of synthetic feasiblity. Graph models faced chanllenge in generating large molcules (&gt; 100 HAs). Selfies provided advantage here. All of the models seem to generate novel molecules - how practical each of these novel molecules are is yet an open question.</p>
<ul>
<li><a href="https://arxiv.org/abs/1811.12823">MOSES - Benchmarking platform for generative models</a>.</li>
</ul>
<p>Propose a platform to deploy and compare state-of-the-art generative models for exploring molecular space on same dataset. In addition the authors also propose list of metrics to evaluate the quality and diversity of the generated structures.</p>
<ul>
<li><a href="https://www.benevolent.com/guacamol">GuacaMol: Benchmarking models for De Novo Molecular Design</a>. <a href="https://www.benevolent.com/research/guacamol-benchmarking-models-for-de-novo-molecular-design">Blogpost</a></li>
</ul>
<p>Evaluation framework from BenevolentAI to compare different de-novo design models.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.0c01328">J. Zhang, R. Mercado, O. Engkvist, and H. Chen, “Comparative Study of Deep Generative Models on Chemical Space Coverage,” J. Chem. Inf. Model., vol.&nbsp;61, no. 6, pp.&nbsp;2572–2581, Jun.&nbsp;2021.</a></li>
</ul>
<p>Interesting analysis from team at AstraZeneca R&amp;D. They look at the chemical space coverage accounted by the SOTA generative models. Proposes a metric for evaluating space coverage, and thereby comparing different SOTA models, using a reference data (GDB-13 in this case). The new metric computes how much of the GDB-13 dataset can be recovered by a model that is trained on small GDB subset. Generative models were trained on same 1M data points and 1B molecules were then sampled from each model. It was seen that at most 39% of the molecules in the parent dataset were sampled / generated by the model. Most models sampled the same compounds atleast twice. It was observed that graph-based model sampled much diverse molecules than string-based methods. Besides, the coverage of GAN-based models was worse compared to Language and Graph models.</p>
<ul>
<li><a href="https://doi.org/10.1021/acs.jcim.0c00174">Gao, W.; Coley, C. W. The Synthesizability of Molecules Proposed by Generative Models. J. Chem. Inf. Model. 2020</a></li>
</ul>
<p>This paper looks at different ways of integrating synthesizability criteria into generative models.</p>
<ul>
<li><a href="https://github.com/MolecularAI/REINVENT4">REINVENT4: Modern AI–Driven Generative Molecule Design</a> [Supported with PyTorch 2.0]</li>
</ul>
<p>REINVENT is a molecular design tool for de novo design, scaffold hopping, R-group replacement, linker design, molecule optimization, and other small molecule design tasks. At its heart, REINVENT uses a Reinforcement Learning (RL) algorithm to generate optimized molecules compliant with a user defined property profile defined as a multi-component score. Transfer Learning (TL) can be used to create or pre-train a model that generates molecules closer to a set of input molecules.</p>
<p><strong>Genetic algorithms</strong></p>
<ul>
<li><p><a href="https://arxiv.org/abs/2310.09267">Tripp, Austin, and José Miguel Hernández-Lobato. “Genetic algorithms are strong baselines for molecule generation.” arXiv preprint arXiv:2310.09267 (2023).</a></p></li>
<li><p><a href="https://pubs.rsc.org/en/content/articlehtml/2017/sc/c8sc05372c">Jensen, Jan H. “A graph-based genetic algorithm and generative model/Monte Carlo tree search for the exploration of chemical space.” Chemical science 10.12 (2019): 3567-3572.</a>. <a href="https://github.com/AustinT/mol_ga">Code</a></p></li>
</ul>
<p><strong>Language models:</strong></p>
<ul>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/65e718eee9ebbb4db9f21886">PromptSMILES: Prompting for scaffold decoration and fragment linking in chemical language models</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41467-024-53632-4">Wu, K., Xia, Y., Deng, P., Liu, R., Zhang, Y., Guo, H., … &amp; Liu, T. Y. (2024). TamGen: drug design with target-aware molecule generation through a chemical language model. Nature Communications, 15(1), 9360.</a></p></li>
</ul>
<p>Collaboration with Microsoft AI and Global Health Drug Discovery Institute. TamGen, a method that employs a GPT-like chemical language model and enables target-aware molecule generation and compound refinement.The authors identified 7 compounds showing compelling inhibitory activity against the Tuberculosis ClpP protease. The model considers 1-D information of the protein and molecule.</p>
<ul>
<li><a href="https://arxiv.org/pdf/2103.03864.pdf">Maziarz, Krzysztof, et al.&nbsp;“Learning to extend molecular scaffolds with structural motifs.” arXiv preprint arXiv:2103.03864 (2021).</a>. <a href="https://github.com/microsoft/molecule-generation">Github</a></li>
</ul>
<p>Team at Novartis and Microsoft propose MoLeR, graph based model to generate molecule using scaffold as a seed. Scaffold based SAR speed up shown.</p>
<ul>
<li><p><a href="https://www.nature.com/articles/s42256-022-00580-7.epdf?sharing_token=p5m9Z0797IQeBDOiMGn71dRgN0jAjWel9jnR3ZoTv0MeIJPs9pbG9QLaEN_McFTR3KHv1tHh1FDNJB4ZuILdAmRtINVn6KqXrLkPhEiAZW5mM0dWWKSmPk82eibEUBx01sLTSHx6w903cDaUoXg9lAGzcHY_ifmakrBcIzUUDwI%3D">Ross, Jerret, et al.&nbsp;“Large-scale chemical language representations capture molecular structure and properties.” Nature Machine Intelligence 4.12 (2022): 1256-1264.</a>. [Github]https://github.com/IBM/molformer?tab=readme-ov-file)</p></li>
<li><p><a href="https://chemrxiv.org/articles/preprint/Beyond_Generative_Models_Superfast_Traversal_Optimization_Novelty_Exploration_and_Discovery_STONED_Algorithm_for_Molecules_using_SELFIES/13383266">SELFIES and generative models using STONED</a></p></li>
</ul>
<p><a href="https://arxiv.org/pdf/2102.00700.pdf">Reproducibility study of the STONED work from Jablonka et. al.</a></p>
<p>Representation using SELFIES proposed to make it much more powerful</p>
<ul>
<li><a href="https://chemrxiv.org/engage/chemrxiv/article-details/60c7591a9abda2847ff8ea1f">Iovanac, Nicolae C., Robert MacKnight, and Brett Savoie. “Actively Searching: Inverse Design of Novel Molecules with Simultaneously Optimized Properties.” ChemRxiv (2021)</a></li>
</ul>
<p>Using quantum chemistry attributes calculated on-the-fly as scoring functions for sampling the generative model chemical space. Active learning strategy is deployed to explore the area of space where the properties of the molecules are unknown.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acscentsci.7b00572">R. Gómez-Bombarelli et al., “Automatic Chemical Design Using a Data-Driven Continuous Representation of Molecules,” ACS Cent. Sci., vol.&nbsp;4, no. 2, pp.&nbsp;268–276, 2018</a></li>
</ul>
<p>One of the first implementation of a variational auto-encoder for molecule generation.</p>
<p><strong>Graph-based</strong></p>
<ul>
<li><a href="https://arxiv.org/abs/2202.00658?">Flam-Shepherd, Daniel, Alexander Zhigalin, and Alán Aspuru-Guzik. “Scalable Fragment-Based 3D Molecular Design with Reinforcement Learning.” arXiv preprint arXiv:2202.00658 (2022)</a></li>
</ul>
<p>Reinforcement learning-based generative model whici is an update on point cloud approach by the same group to now incorporate ‘grammar’ for building molecules in form of functional groups in 3D space.</p>
<ul>
<li><a href="https://arxiv.org/abs/1802.04364">W. Jin, R. Barzilay, and T. Jaakkola, “Junction tree variational autoencoder for molecular graph generation,” 35th Int. Conf. Mach. Learn. ICML 2018, vol.&nbsp;5, pp.&nbsp;3632–3648, 2018</a></li>
</ul>
<p>Junction tree based decoding. Define a grammar for the small molecule and find sub-units based on that grammar to construct a molecule. The molecule is generated in two-steps: first being generating the scaffold or backbone of the molelcule, then the nodes are added with molecular substructure as identified from the ‘molecular grammar’.</p>
<p><strong>GANs</strong></p>
<ul>
<li><a href="https://arxiv.org/abs/1805.11973">MolGAN: An implicit generative model for small molecular graphs, N. De Cao and T. Kipf, 2018</a></li>
</ul>
<p>Generative adversarial network for finding small molecules using graph networks, quite interesting. Avoids issues arising from node ordering that are associated with likelihood based methods by using an adversarial loss instead (GAN)</p>
<p><strong>Scaffold-retained</strong></p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.1c01130">Kaitoh, Kazuma, and Yoshihiro Yamanishi. “Scaffold-Retained Structure Generator to Exhaustively Create Molecules in an Arbitrary Chemical Space.” Journal of Chemical Information and Modeling (2022).</a></p></li>
<li><p><a href="https://arxiv.org/pdf/2103.03864.pdf">Maziarz, Krzysztof, et al.&nbsp;“Learning to extend molecular scaffolds with structural motifs.” arXiv preprint arXiv:2103.03864 (2021).</a>. <a href="https://github.com/microsoft/molecule-generation">Github</a></p></li>
</ul>
<p>Team at Novartis and Microsoft propose MoLeR, graph based model to generate molecule using scaffold as a seed. Scaffold based SAR speed up shown.</p>
<p><strong>Synthesizable-space aware</strong></p>
<ul>
<li><a href="https://chemrxiv.org/engage/chemrxiv/article-details/67bf3e7efa469535b90be62b">Growing and Linking Optimizers: Synthesis-driven Molecule Design</a></li>
</ul>
<p>The team at Iktos suggest a reaction-based generative design. They break down the method into growing and linking steps to emulate real-life process. They compare their approach to REIVENT and show their model is able to design molecule which are not only synthesizable but also closer to the pre-defined desired properties.</p>
<ul>
<li><a href="https://arxiv.org/abs/2410.03494">Gao, Wenhao, Shitong Luo, and Connor W. Coley. “Generative artificial intelligence for navigating synthesizable chemical space.” arXiv preprint arXiv:2410.03494 (2024).</a></li>
</ul>
<p>SynFormer from Coley group to look at synthesizable space. The idea is to constraint the molecules generated by the transformations amenable to a particular platform, like automated synthesis workflow. Building blocks are selected through a diffusion module. They propose two models - a decoder to get feedback from black-box optimizer and a encoder-decoder system to incorporate synthetic pathways in the design proposal.</p>
<ul>
<li><p><a href="https://arxiv.org/abs/2406.04628">Luo, S., Gao, W., Wu, Z., Peng, J., Coley, C. W., &amp; Ma, J. (2024). Projecting Molecules into Synthesizable Chemical Spaces.&nbsp;<em>arXiv preprint arXiv:2406.04628</em>.</a></p></li>
<li><p><a href="https://pubs.rsc.org/en/content/articlelanding/2024/dd/d3dd00230f">Expanding the chemical space using a chemical reaction knowledge graph</a></p></li>
<li><p><a href="https://www.nature.com/articles/s42256-024-00809-7">Swanson, K., Liu, G., Catacutan, D.B. et al.&nbsp;Generative AI for designing and validating easily synthesizable and structurally novel antibiotics. Nat Mach Intell 6, 338–353 (2024).</a></p></li>
</ul>
<p>SyntheMol to generate reaction-based molecules by choosing reaction and the building blocks to connect them by. They use this approach for finding novel molecules for antibacterial discovery.</p>
<ul>
<li><p><a href="https://onlinelibrary.wiley.com/doi/10.1002/advs.202206674?utm_source=pocket_saves">Seo, Seonghwan, Jaechang Lim, and Woo Youn Kim. “Molecular Generative Model via Retrosynthetically Prepared Chemical Building Block Assembly.” Advanced Science (2023): 2206674.</a></p></li>
<li><p><a href="https://arxiv.org/pdf/2012.11522.pdf">Bradshaw, John, et al.&nbsp;“Barking up the right tree: an approach to search over molecule synthesis dags.” Advances in neural information processing systems 33 (2020): 6852-6866.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.1c00469">Fialková, Vendy, et al.&nbsp;“LibINVENT: reaction-based generative scaffold decoration for in silico library design.” Journal of Chemical Information and Modeling 62.9 (2021): 2046-2063.</a></p></li>
<li><p><a href="https://arxiv.org/pdf/2106.03394.pdf">Nguyen, Dai Hai, and Koji Tsuda. “A generative model for molecule generation based on chemical reaction trees.” arXiv preprint arXiv:2106.03394 (2021).</a></p></li>
</ul>
<p>Authors propose a generative model to generate molecules via multi-step chemical reaction trees, each campaign first generates a reaction-tree with template transformations as breaking points.</p>
<ul>
<li><a href="https://arxiv.org/abs/1906.05221">Bradshaw, John, et al.&nbsp;“A model to search for synthesizable molecules.” Advances in Neural Information Processing Systems 32 (2019).</a></li>
</ul>
<p><strong>Diffusion models</strong></p>
<ul>
<li><a href="https://arxiv.org/pdf/2411.04130v1">Adams, K., Abeywardane, K., Fromer, J., &amp; Coley, C. W. (2024). ShEPhERD: Diffusing shape, electrostatics, and pharmacophores for bioisosteric drug design. arXiv preprint arXiv:2411.04130.</a></li>
</ul>
<p>A SE(3)-equivariant diffusion model for 3D molecule structures and their interaction profile with the target of choice. The authors show their model application for typical drug design tasks including hit diversification, bioisosteric replacement and fragment merging, and ligand hopping. Shepherd is a joint denoising diffusion probabilistic model (DDPM) that learns the joint distribution over 3D molecules (atom types, bond types, coordinates) and their 3D shapes, ESP surfaces, and pharmacophores.</p>
<ul>
<li><a href="https://chemrxiv.org/engage/chemrxiv/article-details/66a70a1301103d79c51b3220">Sako, Masami, Nobuaki Yasuo, and Masakazu Sekijima. “DiffInt: A Pharmacophore-Aware Diffusion Model for Structure-Based Drug Design with Explicit Hydrogen Bond Interaction Guidance.” (2024).</a>. <a href="https://github.com/sekijima-lab/DiffInt">Github</a></li>
</ul>
<p>DiffInt as a novel structure-based approach that explicitly addresses interactions. The model naturally incorporates hydrogen bonds between the protein and ligand by treating them as pseudoparticles.</p>
<ul>
<li><a href="https://arxiv.org/abs/2404.19739v1">Mixed Continuous and Categorical Flow Matching for 3D De Novo Molecule Generation</a></li>
</ul>
<p>This model extends beyond traditional diffusion models by learning to map samples directly from arbitrary distributions, allowing for greater flexibility and application-specific model design. It is achieving remarkable efficiency (&gt;10-fold reduction in inference time) and accuracy in molecule generation.</p>
<ul>
<li><p><a href="https://www.nature.com/articles/s42256-024-00815-9">Igashov, Ilia, et al.&nbsp;“Equivariant 3D-conditional diffusion model for molecular linker design.” Nature Machine Intelligence (2024): 1-11.</a>. <a href="https://iwatobipen.wordpress.com/2024/04/24/generate-new-molecules-from-fragments-with-diffusion-model-cheminformatics-rdkit-difflinker-memo/">Blog from Pen</a></p></li>
<li><p><a href="https://arxiv.org/abs/2210.13695">Schneuing, Arne, et al.&nbsp;“Structure-based drug design with equivariant diffusion models.” arXiv preprint arXiv:2210.13695 (2022)</a>. Blog from Pen. <a href="https://iwatobipen.wordpress.com/2024/03/05/pocket-awaer-structure-generation-diffdec-cheminformatics/">Link</a></p></li>
</ul>
<p><strong>3D conformations-aware</strong></p>
<ul>
<li><p><a href="https://www.sciencedirect.com/science/article/pii/S0959440X23000337">Integrating structure-based approaches in generative molecular design</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.1c01535">Bolcato, Giovanni, Esther Heid, and Jonas Boström. “On the Value of Using 3D Shape and Electrostatic Similarities in Deep Generative Methods.” Journal of chemical information and modeling 62.6 (2022): 1388-1398.</a></p></li>
</ul>
<p>Extension to the fragment-based generative design model (DeepFMPO) using reinforcement learning now incorporating 3D electrostatic similarity in the analysis. Ability to replace fragment with similar 3D shape and electrostatics. ESP_sim <a href="https://www.youtube.com/watch?v=Ka08REoGYvI">tutorial</a> for comparison of electrostatic potential and molecule shape is used for this purpose. The authors find scaffold-hopping bioisoteres for CDK2.</p>
<p><strong>Protein-ligand interactions aware</strong></p>
<ul>
<li><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.2c00177">Zhang, Jie, and Hongming Chen. “De novo molecule design using molecular generative models constrained by ligand–protein interactions.” Journal of Chemical Information and Modeling 62.14 (2022): 3291-3306.</a></li>
</ul>
<p><strong>Linker design</strong></p>
<ul>
<li><p><a href="https://arxiv.org/abs/2210.05274">Igashov, Ilia, et al.&nbsp;“Equivariant 3d-conditional diffusion models for molecular linker design.” arXiv preprint arXiv:2210.05274 (2022).</a></p></li>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/62628b2debac3a61c7debf31">Guo, Jeff, et al.&nbsp;“Link-INVENT: Generative Linker Design with Reinforcement Learning.” (2022).</a>. <a href="https://iwatobipen.wordpress.com/2022/06/04/generate-molecules-with-link-invent-rdkit-rinvent-chemoinformatics/">BlogPost</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.9b01120">Imrie, Fergus, et al.&nbsp;“Deep generative models for 3D linker design.” Journal of chemical information and modeling 60.4 (2020): 1983-1995.</a>. <a href="https://iwatobipen.wordpress.com/2020/05/09/replace-core-with-delinker-rdkit-chemoinformatics-deeplearning/">Blogpost</a></p></li>
</ul>
<p>Interesting work on designing linkers using conformation aware generative design algorithm. Think of it like fragment-growing.</p>
<ul>
<li><a href="https://arxiv.org/abs/2211.02660">Nori, Divya, Connor W. Coley, and Rocío Mercado. “De novo PROTAC design using graph-based deep generative models.” arXiv preprint arXiv:2211.02660 (2022).</a></li>
</ul>
<p><strong>Synthetic-cost aware</strong></p>
<ul>
<li><a href="https://www.nature.com/articles/s43588-024-00639-y">Fromer, Jenna C., and Connor W. Coley. “An algorithmic framework for synthetic cost-aware decision making in molecular design. Sparrow”&nbsp;<em>Nature Computational Science</em>&nbsp;(2024): 1-11.</a>. <a href="https://arxiv.org/abs/2311.02187">Preprint</a>. <a href="https://medium.com/@kapeluha.ann00/cost-aware-prioritization-of-molecules-for-synthesis-with-sparrow-68c453b66f03">Medium blogpost</a></li>
</ul>
<p>SPARROW prioritizes molecules that both have high rewards and can be synthesized in a few steps from cheap starting materials. It is also shown to combine the library-focused and generative design based compounds in one setting depending on the harmony of the proposed synthetic routes and costs.</p>
</section>
<section id="computer-aided-synthesis-planning-casp" class="level2">
<h2 class="anchored" data-anchor-id="computer-aided-synthesis-planning-casp">Computer Aided Synthesis Planning (CASP)</h2>
<p><strong>Reviews:</strong></p>
<ul>
<li><p><a href="https://pubs.rsc.org/en/content/articlelanding/2024/dd/d3dd00252g">Torren-Peraire, Paula, et al.&nbsp;“Models Matter: the impact of single-step retrosynthesis on synthesis planning.” Digital Discovery 3.3 (2024): 558-572.</a></p></li>
<li><p><a href="https://pubs.rsc.org/en/content/articlehtml/2021/xx/d0re00340a?casa_token=XlT3Q35wF_QAAAAA:Rcge3W9WUQher8k9zVMPBUQJsCeOskpUZOvQShiCcVZw127BHRFMRrZL0X6Upa9eHD-KMYPBNTaPww">Thakkar, Amol, et al.&nbsp;“Artificial intelligence and automation in computer aided synthesis planning.” Reaction chemistry &amp; engineering 6.1 (2021): 27-51.</a></p></li>
</ul>
<p>Perspective on the current SOTA of synthesis planning, automation, and reaction optimization in drug discovery and development phases using AI and ML.</p>
<ul>
<li><p><a href="https://www.sciencedirect.com/science/article/abs/pii/S0959943621002959?dgcid=author">Madzhidov, T. I., et al.&nbsp;(2021). “Machine learning modelling of chemical reaction characteristics: yesterday, today, tomorrow.” Mendeleev Communications 31(6): 769-780.</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41570-021-00260-x">Jorner, K., et al.&nbsp;(2021). “Organic reactivity from mechanism to machine learning.” Nature Reviews Chemistry 5(4): 240-255.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/pdf/10.1021/acs.jmedchem.9b02120">Struble, T. J., et al.&nbsp;(2020). “Current and Future Roles of Artificial Intelligence in Medicinal Chemistry Synthesis.” J Med Chem 63(16): 8667-8682</a></p></li>
<li><p><a href="https://arxiv.org/pdf/1906.10223.pdf">The Exploration of Chemical Reaction Networks</a></p></li>
</ul>
<p>Perspective article summarising their position on the current state of research and future considerations on developing better reaction network models. Break down the analysis of reaction networks as into 3 classes (1) Front Open End: exploration of products from reactants (2) Backward Open Start: Know the product and explore potential reactants (3) Start to End: Product and reactant known, explore the likely intermediates.</p>
<p>Nice summary of potential challenges in the field:</p>
<ul>
<li>Validating exploration algorithms on a consistent set of reaction system.</li>
<li>Need to generate a comparative metric to benchmark different algorithms.<br>
</li>
<li>Considering effect of solvents and/or protein embeddings in the analysis</li>
</ul>
<p><strong>Best practices</strong></p>
<ul>
<li><a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/minf.202100119?casa_token=SczoexCIDDUAAAAA:3M-uhrRFMaNRnydOQb-NBhJ1VvocjwZ1_ll--OGP1QwJ-X5Amwt24M586NXxbbgazYn0t-NBAmxS1oFk">Gimadiev, T. R., Lin, A., Afonina, V. A., Batyrshin, D., Nugmanov, R. I., Akhmetshin, T., … &amp; Varnek, A. (2021). Reaction Data Curation I: Chemical Structures and Transformations Standardization. Molecular Informatics, 2100119.</a></li>
</ul>
<p>Article from Varnek group on best practices on processing data for reaction informatics.</p>
<p><strong>Benchmarking</strong></p>
<ul>
<li><p><a href="https://pubs.rsc.org/en/content/articlelanding/2024/dd/d4dd00007b">Hastedt, Friedrich, et al.&nbsp;“Investigating the reliability and interpretability of machine learning frameworks for chemical retrosynthesis.”&nbsp;<em>Digital Discovery</em>&nbsp;3.6 (2024): 1194-1212.</a></p></li>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/621c86f3c3e9da4f737b0047">Genheden S, Bjerrum E. PaRoutes: a framework for benchmarking retrosynthesis route predictions. ChemRxiv. Cambridge: Cambridge Open Engage; 2022</a>. <a href="https://github.com/MolecularAI/PaRoutes">Github</a></p></li>
</ul>
<p>Benchmarking framework for comparing different multi-step retrosynthesis methods from researchers at AstraZeneca R&amp;D. Provides 10k synthetic routes which can be used as a validation set for different methodologies, providing a platform for systematic comparison of different methods being proposed in the community.</p>
<p><strong>Classifying chemical reactions:</strong></p>
<ul>
<li><a href="https://www.nature.com/articles/s42256-020-00284-w">Schwaller, Philippe, et al.&nbsp;“Mapping the space of chemical reactions using attention-based neural networks.” Nature Machine Intelligence 3.2 (2021): 144-152.</a>. <a href="https://github.com/rxn4chemistry/rxnfp">rxnfp - Github</a>. <a href="https://arxiv.org/abs/2012.06051">Preprint</a>. <a href="https://cen.acs.org/physical-chemistry/computational-chemistry/Mapping-reaction-space-machine-learning/99/i5?utm_source=LatestNews&amp;utm_medium=LatestNews&amp;utm_campaign=CENRSS">News Article</a>.</li>
</ul>
<p>Transformer-based model for reaction classification. Compared it with BERT. Besides classification, the work also formalizes the reaction fingerprint generation using the learned representations. The reaction fingerprints are visualized using TMAPS.</p>
<ul>
<li><a href="https://chemrxiv.org/engage/chemrxiv/article-details/6112ac487117507542e68bef">Heid, E; Green, W; Machine learning of reaction properties via learned representations of the condensed graph of reaction. ChemRxiv (2021)</a></li>
</ul>
<p>Reaction classifiction prediction using atom-mapped reaction that are used to generate condensed reaction graphs and passed through a GCN-variant as implemented in chemprop.</p>
<p><strong>Reaction-specific features</strong></p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.1c00975">Heid, E., &amp; Green, W. H. (2021). Machine learning of reaction properties via learned representations of the condensed graph of reaction. Journal of Chemical Information and Modeling, 62(9), 2101-2110.</a></p></li>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/60e358fb379e8d3ba9f92d15">Probst, Daniel, Philippe Schwaller, and Jean-Louis Reymond. “Reaction Classification and Yield Prediction Using the Differential Reaction Fingerprint DRFP.” ChemRxiv (2021)</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/ci5006614">Schneider, N., et al.&nbsp;(2015). “Development of a Novel Fingerprint for Chemical Reactions and Its Application to Large-Scale Reaction Classification and Similarity.” Journal of Chemical Information and Modeling 55(1): 39-53.</a></p></li>
</ul>
<p>Using scrapped US Patent data to classify chemical reactions and deploy various fingerprints and ML models for classification.</p>
<p><strong>Atom mapping:</strong></p>
<ul>
<li><p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10937625/pdf/41467_2024_Article_46364.pdf">Chen, Shuan, et al.&nbsp;“Precise atom-to-atom mapping for organic reactions via human-in-the-loop machine learning.” Nature Communications 15.1 (2024): 2250.</a></p></li>
<li><p><a href="https://onlinelibrary.wiley.com/doi/10.1002/minf.202100138?af=R">Lin, A., et al.&nbsp;(2021). “Atom-to-atom Mapping: A Benchmarking Study of Popular Mapping Algorithms and Consensus Strategies.”</a></p></li>
</ul>
<p>Comparative analysis of different atom-mapping schemes for generating atom-mapped reaction features. Comments on the state of the art methods and their performance on a curated reaction database.</p>
<ul>
<li><a href="https://advances.sciencemag.org/content/7/15/eabe4166">Extraction of organic chemistry grammar from unsupervised learning of chemical reactions</a>. <a href="https://github.com/rxn4chemistry/rxnmapper">RXMapper</a></li>
</ul>
<p>Data-driven atom mapping schemes which uses transformers for learning the context of the chemical reaction. Researchers at IBM trained a flavor of language model based on Transformer architecture and used it to find reaction centers and maps atoms. Shown to be robust compared to other SOTA methods.</p>
<ul>
<li><a href="https://www.nature.com/articles/s41467-019-09440-2">Automatic mapping of atoms across both simple and complex chemical reactions</a></li>
</ul>
<p><strong>Predicting reaction outcomes:</strong></p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/jacs.4c03131">Data-Efficient, Chemistry-Aware Machine Learning Predictions of Diels–Alder Reaction Outcomes</a></li>
</ul>
<p>Researchers use NERF that model electron flow in the reaction alongside bond-enviorment to propose diels-alder chemistry products.</p>
<ul>
<li><a href="https://pubs.rsc.org/en/content/articlepdf/2019/sc/c8sc04228d">C. W. Coley et al., “A graph-convolutional neural network model for the prediction of chemical reactivity,” Chem. Sci., vol.&nbsp;10, no. 2, pp.&nbsp;370–377, 2019.</a></li>
</ul>
<p>Template-free prediction of organic reaction outcomes using graph convolutional neural networks</p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acscentsci.7b00064?ref=acsciiVIdeepchemistry">Prediction of Organic Reaction Outcomes Using Machine Learning, ACS Cent. Sci. 2017</a></p></li>
<li><p><a href="https://pubs.rsc.org/en/content/articlehtml/2021/sc/d0sc04823b">Guan, Y., et al.&nbsp;(2021). “Regio-selectivity prediction with a machine-learned reaction representation and on-the-fly quantum mechanical descriptors.” Chemical Science 12(6): 2198-2208</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.3c00393">Multi-Instance Learning Approach to the Modeling of Enantioselectivity of Conformationally Flexible Organic Catalysts</a></p></li>
</ul>
<p>Conformational sampling and designing of chiral organic catalysts.</p>
<p><strong>Yield prediction</strong></p>
<ul>
<li><a href="https://github.com/gskcheminformatics/SynthCoder/tree/main">Synthcoder</a></li>
</ul>
<p>Synthcoder is a GSK’s mini-platform for creating encoder models. The main focus of SynthCoder is a model creation and validation for organic chemistry, however, the platform can be used also with the regular human language or other text-based problems, e.g., protein sequences.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/full/10.1021/jacs.4c00098">Raghavan, Priyanka, et al.&nbsp;“Incorporating Synthetic Accessibility in Drug Design: Predicting Reaction Yields of Suzuki Cross-Couplings by Leveraging AbbVie’s 15-Year Parallel Library Data Set.” Journal of the American Chemical Society (2024).</a></li>
</ul>
<p>Evaluation of AbbVie’s medicinal chemistry library data set to build machine learning models for the prediction of Suzuki coupling reaction yields.</p>
<ul>
<li><a href="https://dl.acm.org/doi/pdf/10.1145/3589335.3651470">Ma, Yihong, et al.&nbsp;“Are we making much progress? Revisiting chemical reaction yield prediction from an imbalanced regression perspective.” Companion Proceedings of the ACM on Web Conference 2024. 2024.</a></li>
</ul>
<p>Through experiments on real-world datasets, they demonstrate that treating reaction yield prediction as an imbalanced regression problem and incorporating cost-sensitive reweighting methods can significantly improve predictions for underrepresented high-yield reactions.</p>
<ul>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/6509a987ed7d0eccc3d2b2c7">Voinarovska, Varvara, et al.&nbsp;“When yield prediction does not yield prediction: an overview of the current challenges.” (2023</a></p></li>
<li><p><a href="https://pubs.rsc.org/en/content/articlehtml/2023/sc/d2sc06041h">Saebi, Mandana, et al.&nbsp;“On the use of real-world datasets for reaction yield prediction.” Chemical science 14.19 (2023): 4997-5005.</a></p></li>
<li><p><a href="https://www.science.org/doi/10.1126/science.aar5169?url_ver=Z39.88-2003&amp;rfr_id=ori:rid:crossref.org&amp;rfr_dat=cr_pub%20%200pubmed">Predicting reaction performance in C–N cross-coupling using machine learning</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.0c01234">Multilabel Classification Models for the Prediction of Cross-Coupling Reaction Conditions</a></p></li>
<li><p><a href="https://iopscience.iop.org/article/10.1088/2632-2153/abc81d">Schwaller, P., et al.&nbsp;(2021). “Prediction of chemical reaction yields using deep learning.” Machine Learning: Science and Technology 2(1)</a>. <a href="https://rxn4chemistry.github.io/rxn_yields/">Tutorial and blogpost</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acscentsci.9b00576">Schwaller, P., et al.&nbsp;(2019). “Molecular Transformer: A Model for Uncertainty-Calibrated Chemical Reaction Prediction.” ACS Central Science 5(9): 1572-1583.</a></p></li>
<li><p><a href="https://www.science.org/doi/full/10.1126/science.aar5169">Ahneman, D. T., Estrada, J. G., Lin, S., Dreher, S. D., &amp; Doyle, A. G. (2018). Predicting reaction performance in C–N cross-coupling using machine learning. Science, 360(6385), 186-190.</a></p></li>
</ul>
<p>Classic paper, one of the firsts to show modeling reaction yields using ML. A random forest algorithm, was used to predict synthetic reaction performance in multidimensional chemical space with high-throughput experimentation data. Descriptors for components in a palladium-catalyzed Buchwald-Hartwig cross-coupling were computed and used as inputs. The random forest model outperformed linear regression in predictive accuracy, even with sparse training sets and out-of-sample predictions, highlighting its potential for synthetic methodology adoption.</p>
<p><strong>Retrosynthetic routes:</strong></p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.3c01685">Do Chemformers Dream of Organic Matter? Evaluating a Transformer Model for Multistep Retrosynthesis</a></li>
</ul>
<p>Transformer model evaluation for retrosynthesis from AZ folks. Template-free method.</p>
<ul>
<li><a href="https://onlinelibrary.wiley.com/doi/10.1002/minf.202300128">Westerlund, Annie, et al.&nbsp;“Data-driven approaches for identifying hyperparameters in multi-step retrosynthesis.” (2023).</a></li>
</ul>
<p>Meta analysis on the best set of hyperparameters for retrosynthesis routines. Here the authors explore different parameters of the retrosynthesis workflow and their impact on the performance of the route scoping. They propose new set of parameters, other than the default, to assist in improving the odds of the software finding a route for diverse of set of molecules. First of its kind look into an approach to identify such a set.</p>
<ul>
<li><p><a href="https://pubs.rsc.org/en/content/articlelanding/2020/sc/c9sc05704h">Schwaller, P., et al.&nbsp;(2020). “Predicting retrosynthetic pathways using transformer-based models and a hyper-graph exploration strategy.” Chemical Science 11(12): 3316-3325.</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41586-020-2855-y">Computational planning of the synthesis of complex natural products</a></p></li>
<li><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-018-0323-6">Watson, I. A., et al.&nbsp;(2019). “A retrosynthetic analysis algorithm implementation.” J Cheminform 11(1)</a></p></li>
<li><p><a href="https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/chem.201605499">Segler, Marwin HS, and Mark P. Waller. “Neural‐symbolic machine learning for retrosynthesis and reaction prediction.” Chemistry–A European Journal 23.25 (2017): 5966-5971.</a></p></li>
</ul>
<p>Hybrid neural-symbolic approach for both retrosynthesis and reaction prediction that can be trained with large reaction sets from databases. Template extraction from known reaction datasets to classify new reaction to known reaction classes.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.1c01065">Seidl, Philipp, et al.&nbsp;“Improving Few-and Zero-Shot Reaction Template Prediction Using Modern Hopfield Networks.” Journal of chemical information and modeling 62.9 (2022): 2111-2120.</a></li>
</ul>
<p>Introduce a template-based single-step retrosynthesis model based on Modern Hopfield Networks, which learn an encoding of both molecules and reaction templates in order to predict the relevance of templates for a given molecule. The model does not consider templates as distinct categories, but can leverage structural information about the template. The retrieval approach enables generalization across templates, which makes zero-shot learning possible and improves few-shot learning. On the single-step retrosynthesis benchmark USPTO-50k, the MHN model reaction reaches the state-of-the-art at top-k accuracy for k ≥ 3.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.2c00321">Tu, Zhengkai, and Connor W. Coley. “Permutation invariant graph-to-sequence model for template-free retrosynthesis and reaction prediction.” Journal of Chemical Information and Modeling (2021).</a></li>
</ul>
<p>Graph2SMILES, a template-free retrosynthesis model to predict reaction outcomes and retrosynthesis routes. This model eliminates the need for any input-side SMILES augmentation, while achieving noticeable improvements over Transformer baselines (especially for top-1 accuracy).</p>
<p><strong>Generate reaction networks:</strong></p>
<ul>
<li><p><a href="https://pubs.rsc.org/en/content/articlehtml/2018/me/c7me00107j">Fooshee, David, et al.&nbsp;“Deep learning for chemical reaction prediction.” Molecular Systems Design &amp; Engineering 3.3 (2018): 442-452.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.0c01480">M. Liu et al., “Reaction Mechanism Generator v3.0: Advances in Automatic Mechanism Generation,” J. Chem. Inf. Model., May 2021</a></p></li>
</ul>
<p>Newest version of RMG (v3) is updated to Python v3. It has ability to generate heterogeneous catalyst models, uncertainty analysis to conduct first order sensitivity analysis. RMG dataset for the thermochemical and kinetic parameters have been expanded.</p>
<ul>
<li><a href="https://s3-eu-west-1.amazonaws.com/itempdf74155353254prod/13076087/More_and_Faster__Simultaneously_Improving_Reaction_Coverage_and_Computational_Cost_in_Automated_Reaction_Prediction_Task_v1.pdf">More and Faster: Simultaneously Improving Reaction Coverage and Computational Cost in Automated Reaction Prediction Tasks</a></li>
</ul>
<p>Presents an algorithmic improvement to the reaction network prediction task through their YARP (Yet Another Reaction Program) methodology. Shown to reduce computational cost of optimization while improving the diversity of identified products and reaction pathways.</p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/abs/10.1021/acscentsci.9b00576">Molecular Transformer: A Model for Uncertainty-Calibrated Chemical Reaction Prediction</a></p>
<ul>
<li>Follow-up: <a href="https://www.nature.com/articles/s41467-021-21895-w">Quantitative interpretation explains machine learning models for chemical reaction prediction and uncovers bias</a></li>
</ul></li>
<li><p><a href="https://chemrxiv.org/articles/preprint/Automatic_discovery_of_chemical_reactions_using_imposed_activation/13008500/1">Automatic discovery of chemical reactions using imposed activation</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41467-020-19267-x">Machine learning in chemical reaction space</a></p></li>
</ul>
<p>Look at exploration of reaction space rather than compound space. SOAP kernel for representing the moelcules. Estimate atomization energy for the molecules using ML. Calculate the d(AE) for different ML-estimated AEs. Reaction energies (RE) are estimated and uncertainty propogation is used to estimate the errors. Uncorrelated constant error propogation. 30,000 bond breaking reaction steps Rad-6-RE network used. RE prediction is not as good as AE.</p>
<p><strong>Estimate molecular synthesizability</strong></p>
<p>The idea of estimating whether a molecule is ‘synthesizable’ can be thought of from two areas: 1. Complexity based - compare the fragments in the molecule to the known fragments in the chemical space<br>
2. Full retrosynthesis based - entire route is considered for molecule generation. Reactant complexity drives route complexity.</p>
<ul>
<li><a href="https://pubs.rsc.org/en/content/articlehtml/2024/dd/d3dd00219e">Li, Junren, Lei Fang, and Jian-Guang Lou. “Retro-BLEU: quantifying chemical plausibility of retrosynthesis routes through reaction template sequence analysis.” Digital Discovery (2024).</a></li>
</ul>
<p>Retro-BLEU, a statistical metric adapted from the well-established BLEU score in machine translation, to evaluate the plausibility of retrosynthesis routes based on reaction template sequences analysis. The authors use PaRoute to validate this approach.</p>
<ul>
<li><a href="https://jcheminf.biomedcentral.com/articles/10.1186/1758-2946-1-8">Ertl, Peter, and Ansgar Schuffenhauer. “Estimation of synthetic accessibility score of drug-like molecules based on molecular complexity and fragment contributions.” Journal of cheminformatics 1.1 (2009): 1-11.</a>. <a href="https://github.com/rdkit/rdkit/tree/master/Contrib/SA_Score">RDkit implementation</a></li>
</ul>
<p>Synthetic Accessbility score (SA_Score) is a popular heuristic score for quantifying synthesizability. It computes a score using a fragment-contribution approach, where rarer fragments (as judged by their abundance in the PubChem database of 1mil representative cmpds) are taken as an indication of lower synthesizability.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.7b00622">Coley, Connor W., et al.&nbsp;“SCScore: synthetic complexity learned from a reaction corpus.” Journal of chemical information and modeling 58.2 (2018): 252-261.</a>. <a href="https://github.com/deepchem/deepchem/blob/master/examples/tutorials/Synthetic_Feasibility_Scoring.ipynb">DeepChem implementation</a></li>
</ul>
<p>SCScore is a learned synthetic complexity score computed as a neural network model trained on reaction data from the Reaxys database. It was designed with synthesis planning in mind to operate on molecules resembling not just drug-like products but intermediates and simpler building blocks as well.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.1c01476">Liu, Cheng-Hao, et al.&nbsp;“RetroGNN: Fast Estimation of Synthesizability for Virtual Screening and De Novo Design by Learning from Slow Retrosynthesis Software.” Journal of Chemical Information and Modeling 62.10 (2022): 2293-2300.</a></li>
</ul>
<p>RetroGNN is a graph neural network based model to predict outcome of a synthesis planner given the target molecule. Shown to better perform than SAScore. Code is yet to be released.</p>
<ul>
<li><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-024-00879-0">Chen, Shuan, and Yousung Jung. “Estimating the synthetic accessibility of molecules with building block and reaction-aware SAScore.” Journal of Cheminformatics 16 (2024).</a></li>
</ul>
<p>Authors introduce BR-SAScore, an enhanced version of SAScore that integrates the available building block information (B) and reaction knowledge (R) from synthesis planning programs into the scoring process. The score can also identify fragment contributing to the synthetic infeasibility.</p>
<ul>
<li><a href="https://link.springer.com/article/10.1186/s13321-023-00742-8">Parrot, Maud, et al.&nbsp;“Integrating synthetic accessibility with AI-based generative drug design.” Journal of Cheminformatics 15.1 (2023): 83.</a></li>
</ul>
<p>From team at Iktos for triaging molecule designs. The group introduces (retro-score) RScore and RSPred (derived score from RScore using NN). RScore is computed through a full retrosynthesis analysis. The R2 value for RSPred is 0.75.</p>
<section id="data-driven-chemistry-modeling-and-reaction-optimization" class="level3">
<h3 class="anchored" data-anchor-id="data-driven-chemistry-modeling-and-reaction-optimization">Data-driven chemistry modeling and reaction optimization</h3>
<p><strong>Review / Perspectives</strong></p>
<ul>
<li><a href="https://pubs.acs.org/doi/full/10.1021/acscentsci.3c01163#">Raghavan, Priyanka, et al.&nbsp;“Dataset design for building models of chemical reactivity.” ACS Central Science 9.12 (2023): 2196-2204.</a></li>
</ul>
<p>Authors discuss the design of reaction datasets in ways that are conducive to data-driven modeling, emphasizing the idea that training set diversity and model generalizability rely on the choice of molecular or reaction representation. They lay down the experimental constraints associated with generating common types of chemistry datasets and how these considerations should influence dataset design and model building.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.joc.3c00844">Maloney, Michael P., et al.&nbsp;“Negative Data in Data Sets for Machine Learning Training.” Organic Letters (2023).</a></li>
</ul>
<p>Thoughts from industry practioners on how to label low/no yield reactions in electronic lab notebooks (eLNs). This is important when building ML model for reaction outcomes.</p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acscentsci.1c00535">Williams, Wendy L., et al.&nbsp;“The evolution of data-driven modeling in organic chemistry.” ACS central science 7.10 (2021): 1622-1637.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.3c00577">Machine Learning Strategies for Reaction Development: Toward the Low-Data Limit</a></p></li>
</ul>
<p><strong>Industrial reactions commentary</strong></p>
<ul>
<li><p><a href="https://www.nature.com/articles/nrd.2018.116">Expanding the medicinal chemistry synthetic toolbox</a></p></li>
<li><p><a href="https://doi.org/10.1021/jm200187y">The Medicinal Chemist’s Toolbox: An Analysis of Reactions Used in the Pursuit of Drug Candidates</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/jacs.4c00807?ref=recommended">Late-Stage Saturation of Drug Molecules</a></p></li>
</ul>
<p><strong>Substrate Scoping</strong></p>
<p>Area to understand the coverage of chemical space by a specific reaction transformation. Knowing which substrates can be used for a specific type of reactions can accelerate the generation of HTE datasets, and also reduce wastage and failures in searching for right substrates. Every new reaction protocol which is proposed would have a corresponding set of amenable ‘action-space’ for the ligands.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/full/10.1021/jacs.5c01120">Revealing the Relationship between Publication Bias and Chemical Reactivity with Contrastive Learning</a></li>
</ul>
<p>Coley group explore publication bias in generating substrate scope for known synthetic methods. They devise a substrate scope contrastive learning method that treat reported substrates as +ve samples and unreported as -ve samples. Using an embedding space trained on substrate publication history the model groups them in two classes. They show the learned embeddings correlate with phys-org reactivity descriptiors.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acscentsci.3c01638">Rana, D., Pflüger, P. M., Hölter, N. P., Tan, G., &amp; Glorius, F. (2024). Standardizing Substrate Selection: A Strategy toward Unbiased Evaluation of Reaction Generality. ACS Central Science, 10(4), 899-906.</a></li>
</ul>
<p>The authors report a standardized substrate selection strategy which mitigates biases found in traditional substrate scoping tables. This way the chemists can showcase unbiased applicability of novel methodologies facilitating their practical applications.</p>
<ul>
<li><p><a href="https://arxiv.org/pdf/2402.16882">Gao, W., Raghavan, P., Shprints, R., &amp; Coley, C. W. (2024). Substrate Scope Contrastive Learning: Repurposing Human Bias to Learn Atomic Representations. arXiv preprint arXiv:2402.16882.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/jacs.1c12203">Kariofillis, Stavros K., et al.&nbsp;“Using data science to guide aryl bromide substrate scope analysis in a Ni/photoredox-catalyzed cross-coupling with acetals as alcohol-derived radical sources.” Journal of the American Chemical Society 144.2 (2022): 1045-1055.</a></p></li>
</ul>
<p>Integration of data science techniques, including DFT featurization, dimensionality reduction, and hierarchical clustering, to delineate a diverse and succinct collection of aryl bromides that is representative of the chemical space of the substrate class</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.orglett.2c03246">On the Topic of Substrate Scope</a></li>
</ul>
<p><strong>Articles</strong></p>
<ul>
<li><a href="https://pubs.rsc.org/en/content/articlelanding/2022/re/d2re00008c">Pomberger, Alexander, et al.&nbsp;“The effect of chemical representation on active machine learning towards closed-loop optimization.” Reaction Chemistry &amp; Engineering 7.6 (2022): 1368-1379.</a></li>
</ul>
<p>Lapkin and co look at the effect of chemical representation on reaction performance and condition prediction tasks. They look at the high throughput experientation generated datasets and the impact of calculated chemical descriptors on the prediction of reaction yields. They show tailored descriptions did not outperform the traditional ones but larger initial data accelerated reaction performance.</p>
<ul>
<li><a href="https://chemrxiv.org/engage/chemrxiv/article-details/65d79de59138d23161bec6e6">Haas, Brittany, et al.&nbsp;“Rapid Prediction of Conformationally-Dependent DFT-Level Descriptors using Graph Neural Networks for Carboxylic Acids and Alkyl Amines.” (2024).</a></li>
</ul>
<p>2D and 3D-aware GNNs to predict DFT descriptors for conformationally flexible molecules, focusing on carboxylic acid and amines in particular.</p>
<ul>
<li><a href="https://www.nature.com/articles/s41586-024-07021-y#citeas">Wang, J.Y., Stevens, J.M., Kariofillis, S.K. et al.&nbsp;Identifying general reaction conditions by bandit optimization. Nature 626, 1025–1033 (2024).</a></li>
</ul>
<p>Latest from Abigail Doyle’s group where they use bandit optimization routine, related to Thompson sampling, to find reaction condition.</p>
<ul>
<li><a href="https://www.science.org/doi/full/10.1126/sciadv.adj2314">Götz, Julian, et al.&nbsp;“High-throughput synthesis provides data for predicting molecular properties and reaction success.” Science advances 9.43 (2023)</a>. <a href="https://github.com/jugoetz/slap-platform-predict">Github</a></li>
</ul>
<p>The authors propose a platform that is built for looking at photocatalytic N-heterocycle synthesis connected with HTE, automated purification, and physicochemical assays. Implement train-test split in 3 different strategies to minimize ligand overlap.</p>
<ul>
<li><p><a href="https://pubs.rsc.org/en/Content/ArticleLanding/2024/SC/D3SC05353A">Machine learning from quantum chemistry to predict experimental solvent effects on reaction rates</a></p></li>
<li><p><a href="https://chemistry-europe.onlinelibrary.wiley.com/doi/full/10.1002/chem.202301957">Casetti, Nicholas, et al.&nbsp;“Combining Molecular Quantum Mechanical Modeling and Machine Learning for Accelerated Reaction Screening and Discovery.” Chemistry–A European Journal 29.60 (2023): e202301957.</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41586-021-03213-y">B. J. Shields et al., “Bayesian reaction optimization as a tool for chemical synthesis,” Nature, vol.&nbsp;590, no. June 2020, p.&nbsp;89, 2021</a>. <a href="https://github.com/b-shields/edbo">Github</a></p></li>
</ul>
<p>Experimental design using Bayesian Optimization. Look at 3 rxn class with multiple reaction parameters - temp solvent ligand. Algorithm identifies the optimal conditions. Variables looked into: ligands, bases, solvents, temperatures, concentrations. Algorithm arrived at 99% yields consistently - which was possible by using unusual ligand not known to work well (cognitive bias).</p>
<ul>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/62f6966269f3a5df46b5584b">Torres, Jose Garrido, et al.&nbsp;“A Multi-Objective Active Learning Platform and Web App for Reaction Optimization.” (2022).</a>. <a href="https://pubs.acs.org/doi/full/10.1021/jacs.2c08592?ref=recommended">Follow-up Version 2.0</a></p></li>
<li><p><a href="https://arxiv.org/abs/2203.17241">Hickman, Riley J., et al.&nbsp;“Bayesian optimization with known experimental and design constraints for chemistry applications.” arXiv preprint arXiv:2203.17241 (2022).</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/jacs.1c09718?casa_token=SsZZYyjrgFUAAAAA%3AGnpCghTv-1TwIGt3sANEQ_abtAe_sh-wHoltFPY41NP0vgKose0mzPvCpQMVziiZe6I2yBEihBCrSMI6">Gensch, Tobias, et al.&nbsp;“A comprehensive discovery platform for organophosphorus ligands for catalysis.” Journal of the American Chemical Society 144.3 (2022): 1205-1217.</a></p></li>
</ul>
<p>Also called Kraken - a discovery platform covering monodentate organophosphorus(III) ligands providing comprehensive physicochemical descriptors based on representative conformer ensembles. Using quantum-mechanical methods, the authors calculated descriptors for 1558 ligands, including commercially available examples, and trained machine learning models to predict properties of over 300000 new ligands.</p>
<ul>
<li><p><a href="https://aip.scitation.org/doi/abs/10.1063/5.0048164">Häse, Florian, et al.&nbsp;“Gryffin: An algorithm for Bayesian optimization of categorical variables informed by expert knowledge.” Applied Physics Reviews 8.3 (2021): 031406.</a></p></li>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/62b2dc4b7da6ce2ddb1b3264">Dotson, Jordan, et al.&nbsp;“Data-driven multi-objective optimization tactics for catalytic asymmetric reactions.” (2022).</a></p></li>
</ul>
<p>Multi-objective optimization of catalytic reactions that employ chiral bisphosphine ligands. Optimization of 2 sequential reactions in asymmetric synthesis of API. Classification method identify active catalysts – 5% yield (user provided) cutoff for binary classification. Linear regression to model reaction selectivity. DFT-derived descriptor dataset of &gt;550 bisphosphine ligands. Develop an interpretable chemical space mapping tool using PCA. Look at the domain of applicability with the euclidean distance in chemical space.</p>
<p><strong>Generate catalysts</strong></p>
<ul>
<li><a href="https://pubs.rsc.org/en/content/articlelanding/2023/DD/D2DD00125J">Schilter, Oliver, et al.&nbsp;“Designing catalysts with deep generative models and computational data. A case study for Suzuki cross coupling reactions.” Digital Discovery (2023).</a></li>
</ul>
<p>Use VAE and RNN to propose new catalyst for Suzuki cross-coupling reaction. The trained models are used to find catalyst’s binding energy and find high percentage of novel and valid designs.</p>
<p><strong>Databases</strong></p>
<ul>
<li><a href="https://pubs.rsc.org/en/content/articlelanding/2024/dd/d4dd00120f">Avila, Claudio, et al.&nbsp;“Chemistry in a graph: modern insights into commercial organic synthesis planning.” Digital Discovery (2024).</a></li>
</ul>
<p>Team from Pfizer use Graph Datasets and Network visualization to show how process chemistry data (GLP1 inhibitor Lotiglipron in this case) can be stored, queried, and used for illustration purposes. They demonstrate the utility of knowledge graph for optimizing the route selection process. Neo4J is used for querying the dataset.</p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/jacs.1c09820?utm_source=pocket_mylist">Kearnes, S. M., et al.&nbsp;(2021). “The Open Reaction Database.” Journal of the American Chemical Society.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.3c00607">Data Sharing in Chemistry: Lessons Learned and a Case for Mandating Structured Reaction Data</a></p></li>
<li><p><a href="https://www.science.org/doi/10.1126/science.abo0058">Rohrbach, Simon, et al.&nbsp;“Digitization and validation of a chemical synthesis literature database in the ChemPU.” Science 377.6602 (2022): 172-180.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.1c01105">CGRdb2.0: A Python Database Management System for Molecules, Reactions, and Chemical Data</a></p></li>
</ul>
<p><strong>Reaction sanitization</strong></p>
<ul>
<li><p><a href="https://chemrxiv.org/engage/api-gateway/chemrxiv/assets/orp/resource/item/64ca5d3e4a3f7d0c0d78ca42/original/or-derly-datasets-and-benchmarks-for-chemical-reaction-data.pdf">ORDerly: Datasets and benchmarks for chemical reaction data</a></p></li>
<li><p><a href="https://onlinelibrary.wiley.com/doi/full/10.1002/minf.202100119">Reaction Data Curation I: Chemical Structures and Transformations Standardization</a></p></li>
</ul>
<p><strong>Reaction data extraction</strong></p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.jcim.3c01281">Dong, Qingyang, and Jacqueline M. Cole. “Snowball 2.0: Generic Material Data Parser for ChemDataExtractor.” Journal of Chemical Information and Modeling (2023).</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.3c00422">ReactionDataExtractor 2.0: A Deep Learning Approach for Data Extraction from Chemical Reaction Schemes</a></p></li>
</ul>
</section>
<section id="automated-chemistry-workflows" class="level3">
<h3 class="anchored" data-anchor-id="automated-chemistry-workflows">Automated chemistry workflows</h3>
<p><strong>Reviews</strong></p>
<ul>
<li><a href="https://chemrxiv.org/engage/chemrxiv/article-details/65a887f29138d231612bf6df">Self-Driving Laboratories for Chemistry and Materials Science</a></li>
</ul>
<p>This review article provides an in-depth analysis of the state-of-the-art in SDL technology, its applications across various scientific disciplines, and the potential implications for research, and industry. This review additionally provides an overview of the enabling technologies for SDLs, including their hardware, software, and integration with laboratory infrastructure. Most importantly, this review explores the diverse range of scientific domains where SDLs have made significant contributions, from drug discovery and materials science to genomics and chemistry.</p>
<ul>
<li><p><a href="https://pubs.acs.org/doi/abs/10.1021/acs.accounts.2c00220">Seifrid, Martin, et al.&nbsp;“Autonomous Chemical Experiments: Challenges and Perspectives on Establishing a Self-Driving Lab.” Accounts of Chemical Research (2022): e0229862-131.</a></p></li>
<li><p><a href="https://www.sciencedirect.com/science/article/pii/S135964461300069X">Godfrey, Alexander G., Thierry Masquelin, and Horst Hemmerle. “A remote-controlled adaptive medchem lab: an innovative approach to enable drug discovery in the 21st Century.” Drug Discovery Today 18.17-18 (2013): 795-802.</a></p></li>
</ul>
<p>Account of Eli Lilly and Company’s ASL (Automated Synthesis Lab)</p>
<p><strong>Articles</strong></p>
<ul>
<li><p><a href="https://www.science.org/doi/10.1126/science.adi1407">Autonomous, multiproperty-driven molecular discovery: From predictions to measurements and back</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acscentsci.2c00207">Nambiar, Anirudh MK, et al.&nbsp;“Bayesian Optimization of Computer-Proposed Multistep Synthetic Routes on an Automated Robotic Flow Platform.” ACS Central Science (2022).</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/full/10.1021/acs.accounts.0c00674">Wilbraham, Liam, S. Hessam M. Mehr, and Leroy Cronin. “Digitizing chemistry using the chemical processing unit: from synthesis to discovery.” Accounts of Chemical Research 54.2 (2020): 253-262.</a></p></li>
</ul>
</section>
<section id="dna-encoded-libraries" class="level3">
<h3 class="anchored" data-anchor-id="dna-encoded-libraries">DNA-encoded Libraries</h3>
<ul>
<li><a href="https://www.nature.com/articles/nchembio.211">Matthew Clark, et. al.&nbsp;DNA-encoded small-molecule libraries (DEL)</a>. <a href="https://cen.acs.org/articles/95/i25/DNA-encoded-libraries-revolutionizing-drug.html">C&amp;EN article on the topic</a></li>
</ul>
<p>New form of storing huge amounts of molecule related data using DNA. Made partially possible by low cost of DNA sequencing. Each molecule in the storage is attached with a DNA strand which encode information about its recipe.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/abs/10.1021/acs.jmedchem.0c00452">Follow up to the work with Machine Learning for hit finding.</a></li>
</ul>
<p>DNA encodings for discovery of novel small-molecule protein inhibitors. Outline a process for building a ML model using DEL. Compare graph convolutions to random forest for classification tasks with application to protein target binding. Graph models seemed to achieve high hit rate comapred to random forest. Apply diversity, logistical, structural filtering to search for novel candidates. First work to use GCN for hit searching.</p>
<ul>
<li><p><a href="https://www.nature.com/articles/s42004-020-00374-1?error=cookies_not_supported&amp;code=2d1394f8-2e1b-46ef-b926-9441292aea56">Martín, A., et al.&nbsp;(2020). “Navigating the DNA encoded libraries chemical space.” Communications Chemistry 3(1).</a></p></li>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/60fac8718804431689e3155b">Zabolotna, Y., Pikalyova, R., Volochnyuk, D., Horvath, D., Marcou, G., &amp; Varnek, A. (2021). Exploration of the chemical space of DNA-encoded libraries.</a></p></li>
<li><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.2c01608">Shmilovich, Kirill, et al.&nbsp;“DEL-Dock: Molecular Docking-Enabled Modeling of DNA-Encoded Libraries.” arXiv preprint arXiv:2212.00136 (2022).</a></p></li>
</ul>
<p>Propose a way to incoporate 3D-spatial information in the DEL read outs to denoise the data.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.3c00588">Zhang, Chris, et al.&nbsp;“Building Block-Based Binding Predictions for DNA-Encoded Libraries.” (2023).</a> <a href="https://github.com/MobleyLab/DEL_analysis">Github</a></li>
</ul>
<p>Set of informatic tools to look at BBs producitivity in DEL screens and guide designs for new DELs. Authors calculate joint probabilities of the BBs for its activity and find increasing binding metric for individual BBs also increases the overall binding energy. The authors then cluster these BBs using 2D and 3D tanimoto FPs (3D Tanimoto Combo) and HDBSCAN clustering. Good workflow for implementing 3D-based ROCs filtering.</p>
</section>
</section>
<section id="large-language-models-llms" class="level2">
<h2 class="anchored" data-anchor-id="large-language-models-llms">Large Language Models (LLMs)</h2>
<p>It’s a stretch to say that GPT-4 or any other LLM understands Chemistry.</p>
<p>At this point, LLMs seem to have two general use cases. First, summarization and information retrieval. LLMs can parse vast collections of text, which can be queried using natural language. These information retrieval capabilities have many applications, from writing computer code and collating clinical trial results to summarizing papers on a specific topic.</p>
<p>While there are still issues with LLMs hallucinating and providing incorrect information, tools and strategies are being developed to ensure the validity of LLM responses.</p>
<p>The other area where LLMs appear to be making inroads is workflow management or tools orchestration. Many activities in drug discovery, whether computational or experimental, require long sequences of steps, which can be tedious to orchestrate. These include asking questions about data, analyze results, do routine post processing for comparing with known state of the project.</p>
<p>While it is often possible to script the execution of these steps, scripting requires a detailed knowledge of each step. LLMs have the potential to simplify this process and carry out multi-step procedures given only a set of initial conditions and a final objective. While the amount of progress the field has made in a short time is impressive, I don’t see LLMs replacing scientists any time soon.</p>
<p>Previously the field has propose assistants for this job <a href="https://www.science.org/content/blog-post/lilly-s-virtual-med-chem-assistant">here</a> which comprised of pre-scripted set of rules and processes. While tedious, they seem to add lot of value to project teams for quickly analyzing the SAR. The hope is LLMs might make the dream of all encompasing assistant a reality.</p>
<ul>
<li><p><a href="https://arxiv.org/abs/2402.06196">Total survey of the LLM landscape</a></p></li>
<li><p><a href="https://arxiv.org/pdf/2402.01680">General LLM-based multi-agent survey</a></p></li>
</ul>
<p><strong>Reviews</strong></p>
<ul>
<li><p><a href="https://arxiv.org/abs/2407.01603">Ramos, Mayk Caldas, Christopher J. Collison, and Andrew D. White. “A Review of Large Language Models and Autonomous Agents in Chemistry.”&nbsp;<em>arXiv preprint arXiv:2407.01603</em>&nbsp;(2024).</a>. <a href="https://github.com/ur-whitelab/LLMs-in-science">Github</a></p></li>
<li><p><a href="https://arxiv.org/pdf/2401.14656">Scientific Large Language Models: A Survey on Biological &amp; Chemical Domains</a></p></li>
<li><p><a href="https://arxiv.org/abs/2310.06083">Bran, Andres M., and Philippe Schwaller. “Transformers and Large Language Models for Chemistry and Drug Discovery.” arXiv preprint arXiv:2310.06083 (2023).</a></p></li>
<li><p><a href="https://www.cell.com/cell/fulltext/S0092-8674(24)01070-5">Gao, S., Fang, A., Huang, Y., Giunchiglia, V., Noori, A., Schwarz, J. R., Ektefaie, Y., Kondic, J., &amp; Zitnik, M. (2024). Empowering biomedical discovery with AI agents. Cell, 187(22), 6125–6151. https://doi.org/10.1016/j.cell.2024.09.022</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41524-025-01538-0">Pyzer-Knapp, Edward O., et al.&nbsp;“Foundation models for materials discovery–current state and future directions.” npj Computational Materials 11.1 (2025): 61.</a></p></li>
</ul>
<p><strong>Agents</strong></p>
<ul>
<li><p><a href="https://github.com/snap-stanford/BioDiscoveryAgent">BioDiscoveryAgent</a></p></li>
<li><p><a href="https://github.com/mims-harvard/TxAgent/tree/main">TxAgent</a></p></li>
</ul>
<p>AI Agent for therapeutic reasoning across a universe of tools. I like the Tool-RAG and the plethora of tools that are documented and exposed.</p>
<ul>
<li><a href="https://arxiv.org/abs/2312.07559">PaperQA: Retrieval-Augmented Generative Agent for Scientific Research</a></li>
</ul>
<p>PaperQA, a Retrieval-Augmented Generation (RAG) agent for the scientific literature. PaperQA begins by constructing LLM search queries from a set of keywords. The results of these searches are aggregated into a vector database and combined with a pre-trained LLM to create a summary of the search results. In benchmark comparisons, the differences between answers provided by PaperQA and human evaluators were similar to differences between individual human evaluators. Encouragingly, unlike many other LLMs, PaperQA didn’t hallucinate citations.</p>
<ul>
<li><a href="https://arxiv.org/abs/2304.05376">Bran, Andres M., et al.&nbsp;“ChemCrow: Augmenting large-language models with chemistry tools.” arXiv preprint arXiv:2304.05376 (2023).</a></li>
</ul>
<p>ChemCrow provides software tools for performing domain-specific tasks, including web searches, file format conversions, and similarity searches. Compared with GPT-4, ChemCrow provided superior performance on tasks like synthetic route planning. The authors also point to potential misuse of LLMs and suggest mitigation strategies.</p>
<ul>
<li><a href="https://arxiv.org/abs/2304.05332">Boiko, Daniil A., Robert MacKnight, and Gabe Gomes. “Emergent autonomous scientific research capabilities of large language models.” arXiv preprint arXiv:2304.05332 (2023).</a>. <a href="https://www.nature.com/articles/s41586-023-06792-0">Peer-review</a></li>
</ul>
<p>Coscientist, a set of LLMs for designing and executing organic syntheses. Coscientist consists of four components designed to search the web, write Python code, extract information from documentation, and program laboratory robotics. The authors test Coscientist using several open and closed-source LLMs and present examples of the system’s ability to plan and execute simple organic syntheses.</p>
<ul>
<li><p><a href="https://github.com/lamm-mit/ProtAgents">ProtAgents</a></p></li>
<li><p><a href="https://arxiv.org/abs/2406.11200">Wu, Shirley, et al.&nbsp;“AvaTaR: Optimizing LLM Agents for Tool Usage via Contrastive Reasoning.” The Thirty-eighth Annual Conference on Neural Information Processing Systems.</a></p></li>
</ul>
<p>Framework that optimizes an LLM agent to use the provided tools. This framework is integrated in <a href="https://github.com/stanfordnlp/dspy">DSPy</a></p>
<ul>
<li><a href="https://pubs.rsc.org/en/content/articlelanding/2025/sc/d4sc04757e">Zhang, Chonghuan, et al.&nbsp;“SynAsk: unleashing the power of large language models in organic synthesis.” Chemical Science 16.1 (2025): 43-56.</a></li>
</ul>
<p>New ReAct agent from AIChemEco team. No code or data available. LLM with tool use (langchain + qwen 1.2b). Main value prop. are their bespoke tools (section 2.5.5) yield prediction, condition pred. and retrosynthesis model.</p>
<p><strong>Generative Design</strong></p>
<ul>
<li><a href="https://molleo.github.io/">Wang, Haorui, et al.&nbsp;“Efficient Evolutionary Search Over Chemical Space with Large Language Models.”&nbsp;<em>arXiv preprint arXiv:2406.16976</em>&nbsp;(2024).</a></li>
</ul>
<p>Introduce LLMs for conducting evolutionary algorithm searches.</p>
<p><strong>Foundation models</strong></p>
<ul>
<li><p><a href="https://arxiv.org/abs/2311.12410">Nach0</a></p></li>
<li><p><a href="https://arxiv.org/html/2406.06316v1">Tx-LLM</a></p></li>
</ul>
<p>The team is looking at creating an LLM-based predictive model (regression and classification). They show that one large model can be used to predict multiple end points (think of one model used for all ADME endpoints), and it indicates that training on a variety of tasks can improve overall performance (positive transfer learning).</p>
<p>I am glad to see this work as it shows how much information and feature-richness can be encoded within the transformer model, especially in the low-data regime; that said, one caution with this approach is that the models are purely text-based and extremely black box and correlation doesn’t mean causation, more so here since we don’t have good control over features being used to train the model.</p>
<ul>
<li><a href="https://arxiv.org/abs/2502.07527">NatureLM</a></li>
</ul>
<p>Team at microsoft research introduces introduces Nature Language Model (NatureLM), a sequence-based science foundation model designed for scientific discovery across multiple domains like small molecules, proteins, and materials. NatureLM excels in generating and optimizing scientific entities and offers top performance, matching or surpassing state-of-the-art specialist models.</p>
<ul>
<li><p><a href="https://arcinstitute.org/tools/evo">Evo2</a></p></li>
<li><p><a href="https://www.microsoft.com/en-us/research/blog%20exploring-the-structural-changes-driving-protein-function-with-bioemu-1/">BioEmu-1</a></p></li>
<li><p><a href="https://openreview.net/pdf?id=Ya5OHw7lZ8">Sirumalla, S. K., Farina Jr, D. S., Qiao, Z., Di Cesare, D. A., Farias, F. C., O’Connor, M. B., … &amp; Miller, T. Multi-Modal and Multi-Task Transformer for Small Molecule Drug Discovery. In ICML’24 Workshop ML for Life and Material Science: From Theory to Industry Applications.</a></p></li>
</ul>
<p>From Iambic team: 1B-parameter transformer model pre-trained on 2.25 trillion tokens from diverse datasets focused on drug discovery. It details a comprehensive data pipeline for standardizing and processing data from various sources. The model architecture is based on LLaMA-2 and includes advanced features like SwishGLU and Rotary Positional Encoding. The fine-tuned model outperforms strong baselines in assay prediction tasks.</p>
<p><strong>Predictive modeling</strong></p>
<ul>
<li><p><a href="https://www.nature.com/articles/s42004-024-01220-4">The Goldilocks paradigm: comparing classical machine learning, large language models, and few-shot learning for drug discovery applications</a></p></li>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/652e50b98bab5d2055852dde">Jablonka, Kevin Maik, et al.&nbsp;“Leveraging Large Language Models for Predictive Chemistry.” (2023).</a>. <a href="https://www.nature.com/articles/s42256-023-00788-1">Peer-review</a></p></li>
</ul>
<p>Authors show GPT3 based predictive models perform on-par with SOTA with lower data points. Caution is the models are purely text-based and extreme black box and sometimes, while trite, correlation doesnt mean causation might become important here. Finally the fine tuning doesnt do regression on the data in same sense as a linear regression or random forest would do.</p>
<ul>
<li><a href="https://arxiv.org/abs/2406.03686">BindGPT</a></li>
</ul>
<p><strong>Small molecule related tasks</strong></p>
<ul>
<li><p><a href="https://openreview.net/pdf?id=yRrPfKyJQ2">Liu, Shengchao, et al.&nbsp;“Conversational drug editing using retrieval and domain feedback.” The Twelfth International Conference on Learning Representations. 2024.</a></p></li>
<li><p><a href="https://arxiv.org/pdf/2406.06777">Le, Khiem, et al.&nbsp;“MolX: Enhancing Large Language Models for Molecular Learning with A Multi-Modal Extension.” arXiv preprint arXiv:2406.06777 (2024).</a></p></li>
<li><p><a href="https://www.nature.com/articles/s42256-023-00759-6">Liu, Shengchao, et al.&nbsp;“Multi-modal molecule structure–text model for text-based retrieval and editing.” Nature Machine Intelligence 5.12 (2023): 1447-1457.</a></p></li>
<li><p><a href="https://osu-nlp-group.github.io/LLM4Chem/">Yu, Botao, et al.&nbsp;“LlaSMol: Advancing Large Language Models for Chemistry with a Large-Scale, Comprehensive, High-Quality Instruction Tuning Dataset.” arXiv preprint arXiv:2402.09391 (2024).</a></p></li>
<li><p><a href="https://arxiv.org/pdf/2402.17810.pdf">Pei, Qizhi, et al.&nbsp;“BioT5+: Towards Generalized Biological Understanding with IUPAC Integration and Multi-task Tuning.” arXiv preprint arXiv:2402.17810 (2024).</a></p></li>
</ul>
<p>BioT5+ incorporates several novel features: integration of IUPAC names for molecular understanding, inclusion of extensive bio-text and molecule data from sources like bioRxiv and PubChem, the multi-task in struction tuning for generality across tasks, and a novel numerical tokenization technique for improved processing of numerical data.</p>
<ul>
<li><p><a href="https://github.com/MolecularAI/Chemformer">Irwin, R., Dimitriadis, S., He, J., Bjerrum, E.J., 2021. Chemformer: A Pre-Trained Transformer for Computational Chemistry. Mach. Learn. Sci. Technol.</a>. Previously called <a href="https://github.com/MolecularAI/MolBART">MolBART</a></p></li>
<li><p><a href="https://www.deepmirror.ai/post/one-molecular-generator-to-rule-them-all">LLM-based generation and benchmarking</a></p></li>
<li><p><a href="https://github.com/language-plus-molecules/LPM-24-Tutorial">Language + Molecule Benchmarks</a></p></li>
</ul>
<p><strong>Protein design and mechanics</strong></p>
<ul>
<li><a href="https://www.biorxiv.org/content/10.1101/2024.12.10.627665v1">Queen, Owen, Yepeng Huang, Robert Calef, Valentina Giunchiglia, Tianlong Chen, George Dasoulas, LeAnn Tai et al.&nbsp;“ProCyon: A multimodal foundation model for protein phenotypes.” bioRxiv (2024): 2024-12.</a></li>
</ul>
<p>ProCyon integrates phenotypic and protein data. Authors show its use for identifying protein domains that bind small molecule drugs, predicting peptide binding with enzymes, and assessing the functional impact of Alzheimer’s disease mutations. ProCyon enables conditional retrieval of proteins linked to small molecules through complementary mechanisms of action</p>
<ul>
<li><p><a href="https://www.nature.com/articles/s41587-024-02123-4">Ruffolo, Jeffrey A., and Ali Madani. “Designing proteins with language models.” Nature Biotechnology 42.2 (2024): 200-202.</a></p></li>
<li><p><a href="https://arxiv.org/abs/2402.07148">Buehler, Eric L., and Markus J. Buehler. “X-LoRA: Mixture of Low-Rank Adapter Experts, a Flexible Framework for Large Language Models with Applications in Protein Mechanics and Design.” arXiv preprint arXiv:2402.07148 (2024).</a></p></li>
</ul>
<p>Mixture of LoRA Experts: Leverage the power of fine-tuned LoRA experts by employing a mixture of experts, or MoE technique.</p>
<ul>
<li><a href="https://arxiv.org/abs/2402.04268">Ghafarollahi, Alireza, and Markus J. Buehler. “ProtAgents: Protein discovery via large language model multi-agent collaborations combining physics and machine learning.” arXiv preprint arXiv:2402.04268 (2024).</a></li>
</ul>
<p><strong>Clinical text</strong></p>
<ul>
<li><p><a href="https://www.nature.com/articles/s41746-024-01377-1">Lopez, Ivan, et al.&nbsp;“Clinical entity augmented retrieval for clinical information extraction.” npj Digital Medicine 8.1 (2025): 45.</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41591-024-02855-5">Van Veen, D., Van Uden, C., Blankemeier, L. et al.&nbsp;Adapted large language models can outperform medical experts in clinical text summarization. Nat Med (2024).</a>. <a href="https://github.com/StanfordMIMI/clin-summ">Github</a></p></li>
</ul>
<p>Authors look at clinical summarization and implement quantitative assesments with synctactic, semantic, and conceptual NLP metrics. A clinical reader study with 10 physicians evaluated summary completeness, correctness and conciseness; in most cases, summaries from our best-adapted LLMs were deemed either equivalent (45%) or superior (36%) compared with summaries from medical experts. The research provides evidence of LLMs outperforming medical experts in clinical text summarization across multiple tasks. This suggests that integrating LLMs into clinical workflows could alleviate documentation burden, allowing clinicians to focus more on patient care.</p>
<ul>
<li><a href="https://doi.org/10.48550/arXiv.2404.18416">Saab, K.; et. al.&nbsp;Capabilities of Gemini Models in Medicine. arXiv May 1, 2024.</a></li>
</ul>
<p>Google’s team shows Med-Gemini’s real-world utility by surpassing human experts on tasks such as medical text summarization, alongside demonstrations of promising potential for multimodal medical dialogue, medical research and education.</p>
<p><strong>Data curation</strong></p>
<ul>
<li><a href="https://chemrxiv.org/engage/chemrxiv/article-details/67c6170c6dde43c90858b305">Leong, S. X., Pablo-García, S., Wong, B., &amp; Aspuru-Guzik, A. (2025). MERMaid: Universal multimodal mining of chemical reactions from PDFs using vision-language models.</a></li>
</ul>
<p>Aspuru-Guzik group releases a full suite of tools to ingest, process, and generate knowledge graphs from scientific publications. They show MERMaid, using the vision language models, demonstrate chemical context awareness and reasoning capabilities from the extracted text.</p>
<ul>
<li><a href="https://pubs.rsc.org/en/content/articlehtml/2025/cs/d4cs00913d">From text to insight: large language models for chemical data extraction</a></li>
</ul>
<p>Jablonka group in Jena shared their early findings in using LLMs for chemical and material data extraction tasks. This is structured as a tutotrial review providing specific workflow examples.</p>
<ul>
<li><a href="https://github.com/qai222/LLM_organic_synthesis">Extracting Structured Data from Free-form Organic Synthesis Text</a></li>
</ul>
<p>Hackathon to quickly fine-tune GPT to parse synthesis data and extract relevant chemistry-related information.</p>
<p><strong>Material science</strong></p>
<ul>
<li><a href="https://arxiv.org/abs/2402.04379">Gruver, Nate, et al.&nbsp;“Fine-Tuned Language Models Generate Stable Inorganic Materials as Text.” arXiv preprint arXiv:2402.04379 (2024).</a></li>
</ul>
<p><strong>Reaction development</strong></p>
<ul>
<li><p><a href="https://arxiv.org/abs/2503.08537">Bran, Andres M., Theo A. Neukomm, Daniel P. Armstrong, Zlatko Jončev, and Philippe Schwaller. “Chemical reasoning in LLMs unlocks steerable synthesis planning and reaction mechanism elucidation.” arXiv preprint arXiv:2503.08537 (2025).</a></p></li>
<li><p><a href="https://chemrxiv.org/engage/chemrxiv/article-details/679a8f11fa469535b958a862">Li, H., Sarkar, S., Lu, W., Loftus, P., Qiu, T., Shee, Y., … &amp; Batista, V. (2025). Collective Intelligence of Specialized Language Models Guides Realization of de novo Chemical Synthesis.</a><a href="https://github.com/haoteli/MOSAIC/tree/main">Code</a></p></li>
</ul>
<p>MOSAIC is a fine-tuned LLama 3.1-8b-instruct model for reaction recipe generation.</p>
<ul>
<li><a href="https://arxiv.org/abs/2410.04223">Liu, G., Sun, M., Matusik, W., Jiang, M., &amp; Chen, J. (2024). Multimodal Large Language Models for Inverse Molecular Design with Retrosynthetic Planning. arXiv preprint arXiv:2410.04223.</a></li>
</ul>
<p>The team introduces Llamole, multimodal LLM capable of generating text and graphs.</p>
<p><strong>Agent Engineering</strong></p>
<ul>
<li><p>[HiAgent]https://arxiv.org/abs/2408.09559)</p></li>
<li><p><a href="https://arxiv.org/abs/2502.09858">POPPER</a></p></li>
<li><p><a href="https://arxiv.org/html/2503.00735v3">LADDER</a></p></li>
<li><p><a href="https://arxiv.org/pdf/2412.21154">Narayanan, Siddharth, James D. Braza, Ryan-Rhys Griffiths, Manu Ponnapati, Albert Bou, Jon Laurent, Ori Kabeli et al.&nbsp;“Aviary: training language agents on challenging scientific tasks.” arXiv preprint arXiv:2412.21154 (2024).</a>. <a href="https://github.com/Future-House/aviary">Github</a></p></li>
</ul>
<p>Casting the AI Agentic process as a markov decision process and train language-based AI Agent on biology tasks. They show exceeding human-level performance on SeqQA (dataset consists of MCQs questions, such as counting the fragments after digestion, predicting translated sequences, and identifying polymerase chain reaction primers). The trained Language Agent are compute efficient.</p>
<ul>
<li><a href="https://arxiv.org/abs/2408.08435">Hu, S., Lu, C., &amp; Clune, J. (2024). Automated design of agentic systems. arXiv preprint arXiv:2408.08435.</a><a href="https://www.shengranhu.com/ADAS/">Website</a></li>
</ul>
<p>Framework to automatically creating new agent given appropriate building blocks and combine them in new ways. They proposed Meta-Agent search and show agents can invent novel and power agent designs.</p>
<ul>
<li><a href="https://www.biorxiv.org/content/10.1101/2024.11.11.623004v1">Swanson, Kyle, Wesley Wu, Nash L. Bulaong, John E. Pak, and James Zou. “The Virtual Lab: AI Agents Design New SARS-CoV-2 Nanobodies with Experimental Validation.” bioRxiv (2024): 2024-11.</a></li>
</ul>
<p>A joint paper from James Zou (Stanford) and Chan-Zuckerberg foundation showcases a virtual lab comprising of AI Agentic personas of typical research group collaborating together to design nanobodies. Interesting idea.</p>
<ul>
<li><a href="https://github.com/SakanaAI/AI-Scientist?tab=readme-ov-file">AI Scientist from SakanaAI</a></li>
</ul>
<p>Automation to encompass end-to-end scientific discovery, including coding, experiment execution, and automated peer review for manuscript generation.</p>
<p><strong>Model Improvements</strong></p>
<ul>
<li><a href="https://arxiv.org/abs/2403.09629">Zelikman, E., Harik, G., Shao, Y., Jayasiri, V., Haber, N., &amp; Goodman, N. D. (2024). Quiet-star: Language models can teach themselves to think before speaking. arXiv preprint arXiv:2403.09629.</a></li>
</ul>
</section>
<section id="data-extraction" class="level2">
<h2 class="anchored" data-anchor-id="data-extraction">Data extraction</h2>
<p><strong>Rule-based tool</strong></p>
<ul>
<li><a href="https://www.nextmovesoftware.com/leadmine.html">LeadMine - NextMove</a></li>
</ul>
<p><strong>AI-based tool</strong></p>
<p>Despite so much progress around computer vision and optical character recognition (OCR) the state of the art for molecule image conversion to structure still remains to be manual curation. There have been some interesting tools proposed for automating this using different flavor of computer-vision algorithms.</p>
<ul>
<li><p><a href="https://github.com/DS4SD/MolGrapher">MolGrapher</a></p></li>
<li><p><a href="https://github.com/bayer-science-for-a-better-life/Img2Mol">Img2Mol</a></p></li>
</ul>
<p>One of the core reasons this area has been under explored seems to be molecule patents are MADE to be tough to decipher. The format is non standard and markush enumerations, alongside, their actual chemical space coverage is ill-defined.</p>
<ul>
<li><a href="https://github.com/Kohulan/DECIMER-Image_Transformer">DECIMER</a></li>
</ul>
<p>DECIMER Image Transformer: Deep Learning for Chemical Image Recognition using Efficient-Net V2 + Transformer. <a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00538-8">V1</a>. Extraction of chemical structure through OSCR (Optical chemical structure recognition) from Steinbeck’s group.</p>
<ul>
<li><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.4c00572">Fan, Vincent, et al.&nbsp;“OpenChemIE: An information extraction toolkit for chemistry literature.”&nbsp;<em>Journal of Chemical Information and Modeling</em>&nbsp;(2024).</a></li>
</ul>
<p>Focused on the extraction of reaction data from journals. OpenChemIE is most suited for information extraction on organic chemistry literature, where molecules are generally depicted as planar graphs or written in text and can be consolidated into a SMILES format.</p>
<ul>
<li><a href="https://pubs.rsc.org/en/content/articlelanding/2024/dd/d4dd00091a">Ai, Q., Meng, F., Shi, J., Pelkie, B. G., &amp; Coley, C. W. (2024). Extracting structured data from organic synthesis procedures using a fine-tuned large language model. Digital Discovery.</a></li>
</ul>
<p>Using Llama-2 7b to extract entities from synthesis recipes from reactions.</p>
</section>
<section id="code-packages" class="level2">
<h2 class="anchored" data-anchor-id="code-packages">Code / Packages:</h2>
<p><strong>Peptide Informatics</strong></p>
<p>I visit first-ever RDKit UGM early April 2025 and was impressed by the tools available for manipulating and analyzing peptide chemical space. Below are few:</p>
<ul>
<li><p><a href="https://github.com/rochoa85/PepFun">PepFun</a> - From Novo Nordisk informatics team, they provide functionalities to study peptides at different levels, sequence, structures, and large dataset. It is built on BioPython and RDKit.</p></li>
<li><p><a href="">SynthCoder</a></p></li>
<li><p><a href="https://honegumi.readthedocs.io/en/latest/">Honegumi</a></p></li>
<li><p><a href="https://rxn-insight.readthedocs.io/en/latest/">Rxn-INSIGHT</a></p></li>
<li><p><a href="https://github.com/MolecularAI">Molecular AI department at AstraZeneca R&amp;D</a></p></li>
<li><p><a href="https://www.nature.com/articles/s41598-023-30089-x">Jazzy: Fast calculation of hydrogen-bond strengths and free energy of hydration of small molecules</a></p></li>
<li><p><a href="https://github.com/rinikerlab/GHOST">GHOST: Generalized threshold shifting procedure</a>. <a href="https://pubs.acs.org/doi/10.1021/acs.jcim.1c00160">Paper</a>. <a href="https://greglandrum.github.io/rdkit-blog/exploratory/machinelearning/2021/12/23/ternary-ghost.html#Synthetic-datasets">Blogpost</a></p></li>
</ul>
<p>Automates the selection of decision threshold for imbalanced classification task. The assumption for this method to work is the similar characteristics (like imbalance ratio) of training and test data.</p>
<ul>
<li><a href="https://arxiv.org/abs/1811.12823">MOSES - Benchmarking platform for generative models (PyTorch Implementation)</a>. <a href="https://github.com/molecularsets/moses">Github</a></li>
</ul>
<p>Benchmarking platform to implement molecular generative models. It also provides a set of metrics to evaluate the quality and diversity of the generated molecules. A benchmark dataset (subset of ZINC) is provided for training the models.</p>
<ul>
<li><a href="https://chemrxiv.org/articles/preprint/REINVENT_2_0_an_AI_Tool_for_De_Novo_Drug_Design/12058026/1">Reinvent 4.0 - an AI tool forr de novo drug design</a>. <a href="https://github.com/MolecularAI/Reinvent">Github</a></li>
</ul>
<p>Production-ready tool for de novo design from Astra Zeneca. It can be effectively applied on drug discovery projects that are striving to resolve either exploration or exploitation problems while navigating the chemical space. Language model with SMILE output and trained by “randomizing” the SMILES representation of the input data. Implement reinforcement-leraning for directing the model towards relevant area of interest. Now uses PyTorch 2.0!</p>
<ul>
<li><p><a href="https://chemrxiv.org/articles/OpenChem_A_Deep_Learning_Toolkit_for_Computational_Chemistry_and_Drug_Design/12691943/1">OpenChem</a>. <a href="https://github.com/Mariewelt/OpenChem">Github</a></p></li>
<li><p><a href="https://github.com/deepchem/deepchem">DeepChem (Tensorflow)</a>. <a href="https://deepchem.io">Website</a></p></li>
</ul>
<p>DeepChem aims to provide a high quality open-source toolchain that democratizes the use of deep-learning in drug discovery, materials science, quantum chemistry, and biology - from Github</p>
<ul>
<li><a href="https://github.com/chemprop/chemprop">ChemProp (Pytorch)</a></li>
</ul>
<p>Github repository for implmenting message passing neural networks for molecular property prediction as described in the paper <a href="https://pubs.acs.org/doi/abs/10.1021/acs.jcim.9b00237">Analyzing Learned Molecular Representations for Property Prediction</a> by Yang et. al.&nbsp;</p>
<ul>
<li><p><a href="https://github.com/Bibyutatsu/FastJTNNpy3">FastJTNN - python 3 version of the JT-NN</a></p></li>
<li><p><a href="https://arxiv.org/abs/2003.03123">DimeNet++ - extension of Directional message pasing working (DimeNet)</a>. <a href="https://github.com/klicperajo/dimenet">Github</a></p></li>
<li><p><a href="https://github.com/mjwen/bondnet">BondNet - Graph neural network model for predicting bond dissociation energies, considers both homolytic and heterolytic bond breaking</a>. <a href="https://github.com/mjwen/bondnet">Github</a></p></li>
<li><p><a href="https://duartegroup.github.io/autodE/">AutodE</a></p></li>
<li><p><a href="https://singroup.github.io/dscribe/latest/">DScribe</a></p></li>
<li><p><a href="https://github.com/ReactionMechanismGenerator/RMG-Py">RMG - Reaction Mechanism Generator</a></p></li>
</ul>
<p>Tool to generate chemical reaction networks. Includes Arkane, package for calculating thermodynamics from quantum mechanical calculations.</p>
<ul>
<li><a href="https://pyepal.readthedocs.io/en/latest/?badge=latest">PyePAL</a></li>
</ul>
<p>Active learning approach to efficiently and confidently identify the Pareto front with any regression model that can output a mean and a standard deviation.</p>
<ul>
<li><a href="https://github.com/rxn4chemistry/rxnfp">rxnfp</a></li>
</ul>
<p>Github repository to generate chemical reaction fingerprints from reaction SMILES.</p>
<ul>
<li><a href="https://github.com/cbouy/mols2grid">mols2grid</a></li>
</ul>
<p>Interactive chemical viewer for small molecules (RDKit wrapper)</p>
<ul>
<li><a href="https://github.com/wjm41/molplotly">molplotly</a></li>
</ul>
<p>Spotfire like capabilities to jupyter notebook. Medium article on explaining the MolPlotly. <a href="https://medium.com/@qcojuandavidmarin/molplotly-a-wonderful-tool-for-the-scientist-5ddb9a42c037?utm_source=pocket_saves">Link</a></p>
<ul>
<li><a href="https://github.com/hesther/espsim">ESPsim</a></li>
</ul>
<p>Calculate similarities of shapes and electrostatic potentials between molecules. Pen has a nice blogpost on using to estimate electronic similarities of common bioisosteres. <a href="https://iwatobipen.wordpress.com/2022/12/31/calculate-similarity-of-popular-bioisosters-rdkit-espsim-memo/">blog</a></p>
<ul>
<li><a href="https://github.com/prcurran/hotspots/tree/master?tab=readme-ov-file#scoring">HotSpots: Curran, Peter R., et al.&nbsp;“Hotspots api: a python package for the detection of small molecule binding hotspots and application to structure-based drug design.” Journal of chemical information and modeling 60.4 (2020): 1911-1916.</a></li>
</ul>
<p>Survey protein surfaces for binding hotspots can help to evaluate target tractability and guide exploration of potential ligand binding regions.</p>
<ul>
<li><a href="https://github.com/coleygroup/molpal">MolPal</a></li>
</ul>
<p>Active learning methodology for sampling the chemical space</p>
<ul>
<li><p><a href="https://github.com/GT4SD/gt4sd-core">Generative Toolkit 4 Scientific Discovery</a></p></li>
<li><p><a href="https://github.com/ghiander/chemprop-jazzy">Jazzy + Chemprop</a></p></li>
</ul>
<p>Chemprop version that combines Jazzy (AZ’s workflow for predicting H-bond strength)</p>
</section>
<section id="datasets-chemical-libraries" class="level2">
<h2 class="anchored" data-anchor-id="datasets-chemical-libraries">Datasets &amp; Chemical libraries</h2>
<p><strong>Molecule datasets</strong></p>
<ul>
<li>PubChem: public sourced molecules</li>
<li>ChEMBL: bioactive molecules (most synthetic)</li>
<li>SUREChEMBL: small molecules appearing in Patents</li>
<li>ZINC: collection of synthetic molecules (not all are bioactive)</li>
<li>QM 7/8/9: small molecules having not more than 7/8/9 heavy atoms</li>
<li>GDB-17</li>
<li><a href="https://chemrxiv.org/engage/chemrxiv/article-details/617aa2467a002162403d71f0">Papyrus</a></li>
<li><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00478-9">COCONUT</a>: NP 400k there are some which are not NP</li>
<li><a href="https://mcule.com/database/">Mcule</a>: Used in DEL enumerations</li>
<li><a href="https://go.drugbank.com/">DrugBank</a></li>
</ul>
<p><strong>Reaction Datasets</strong></p>
<ul>
<li>USPTO</li>
<li>Pistachio</li>
<li>Reaxys</li>
<li>Open Reaction Database</li>
</ul>
<p><strong>Commercial (building block) vendors</strong></p>
<ul>
<li>eMolecules</li>
<li>Enamine<br>
</li>
<li>WuXi</li>
<li>Chembridge</li>
<li>Asinex</li>
<li>Molport</li>
<li>Pharmablock</li>
<li>Otava’s CHEMriya</li>
</ul>
</section>
<section id="helpful-utilities" class="level2">
<h2 class="anchored" data-anchor-id="helpful-utilities">Helpful utilities:</h2>
<ul>
<li><p><a href="https://github.com/rdkit/rdkit">RD-Kit</a></p>
<ul>
<li><p><a href="https://colab.research.google.com/drive/16T6ko0YE5WqIRzL4pwW_nufTDn7F3adw">Get Atom Indices in the SMILE:</a></p></li>
<li><p><a href="https://github.com/datamol-org/datamol">Datamol for manipulating RDKit molecules</a></p></li>
</ul></li>
<li><p><a href="https://github.com/molecularsets/moses">MOSES: Molecular generation models benchmark</a></p></li>
<li><p><a href="https://tdcommons.ai">Therapeutics Data Commons</a></p></li>
</ul>
<p>Therapeutics Data Commons is an open-science platform with AI/ML-ready datasets and learning tasks for therapeutics, spanning the discovery and development of safe and effective medicines. TDC also provides an ecosystem of tools, libraries, leaderboards, and community resources, including data functions, strategies for systematic model evaluation, meaning</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>